{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d4f208dc",
   "metadata": {},
   "source": [
    "# 1. Introduction\n",
    "\n",
    "This notebook is for targeted variant analysis of KRAS in circulating tumor DNA (ctDNA) from a single patient (C_fw075). The raw data is from the NCBI BioProject PRJNA714799 related to the following publication:\n",
    "\n",
    "O’Dea et al. Circulating tumor DNA sequencing in colorectal cancer patients treated with first-line chemotherapy with anti-EGFR. Scientific        Reports. 2021\n",
    "\n",
    "\n",
    "You will find sections in the notebook for:\n",
    "\n",
    "- processing raw paired-end ctDNA reads,\n",
    "- align them to the human reference genome GRCh38\n",
    "- perform variant calling with VarScan in the KRAS locus,\n",
    "- annotate and interpret any the mutations in the KRAS region for clinically relevant alterations,\n",
    "- briefly consider whether any new KRAS alterations emerge at follow-up.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc25fdbd",
   "metadata": {},
   "source": [
    "# 2. Environment\n",
    "\n",
    "All analyses were run on WSL2 UBUNTU on a personal (no organization affiliation) Windows 11 machine using a Conda environment named `ngs`. The environment contains common NGS tools:\n",
    "\n",
    "`fastqc` – QC tool\n",
    "`bowtie2` – aligner\n",
    "`samtools` – BAM handling and mpileup generation  \n",
    "`bcftools` – VCF utilities  \n",
    "`varscan` – variant calling  \n",
    "`snpEff` – variant annotation  \n",
    "\n",
    "Python 3.11.14 (Jupyter in VSC) is used for:\n",
    "\n",
    "light bookkeeping (paths, listing files),\n",
    "parsing annotated VCFs and summarizing KRAS variants.\n",
    "Text editing and summarizing.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "65ab00ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "#python imports\n",
    "\n",
    "import pandas as pd\n",
    "import os, glob\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "dd427ee8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "project root: /home/mgpersonal/ctdna_kras_project\n",
      "fastq_dir   : /home/mgpersonal/ctdna_kras_project/data/fastq\n",
      "ref_dir     : /home/mgpersonal/ctdna_kras_project/data/ref\n",
      "bam_dir     : /home/mgpersonal/ctdna_kras_project/results/bam\n",
      "vcf_dir     : /home/mgpersonal/ctdna_kras_project/results/vcf\n",
      "Reference fasta: /home/mgpersonal/ctdna_kras_project/data/ref/Homo_sapiens.GRCh38.dna.primary_assembly.fa\n",
      "Bowtie2 prefix: /home/mgpersonal/ctdna_kras_project/data/ref/GRCh38_primary\n"
     ]
    }
   ],
   "source": [
    "# Define project root and all key subdirectories in one place\n",
    "\n",
    "project_root = Path(\"/home/mgpersonal/ctdna_kras_project\").resolve()\n",
    "\n",
    "data_dir   = project_root / \"data\"\n",
    "fastq_dir  = data_dir / \"fastq\"\n",
    "ref_dir    = data_dir / \"ref\"\n",
    "results_dir = project_root / \"results\"\n",
    "bam_dir     = results_dir / \"bam\"\n",
    "vcf_dir    = results_dir/ \"vcf\"\n",
    "qc_dir  = results_dir/ \"qc\"\n",
    "out_dir=\"${project_root}/results/summary\"\n",
    "\n",
    "# Existing Reference files (Moved a GRCh38 .fa and bowtie2 reference from my system to these directories)\n",
    "ref_fasta  = ref_dir / \"Homo_sapiens.GRCh38.dna.primary_assembly.fa\"\n",
    "ref_prefix  = ref_dir / \"GRCh38_primary\"  # Bowtie2 index prefix\n",
    "\n",
    "# Create directory structure if it doesn't already exist\n",
    "for d in [data_dir, fastq_dir, ref_dir, results_dir, bam_dir, vcf_dir]:\n",
    "    d.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "#Let's make sure everything is prese\n",
    "print(\"project root:\", project_root)\n",
    "print(\"fastq_dir   :\", fastq_dir)\n",
    "print(\"ref_dir     :\", ref_dir)\n",
    "print(\"bam_dir     :\", bam_dir)\n",
    "print(\"vcf_dir     :\", vcf_dir)\n",
    "print(\"Reference fasta:\", ref_fasta)\n",
    "print(\"Bowtie2 prefix:\", ref_prefix)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "844d01f1",
   "metadata": {},
   "source": [
    "# 3. Data Acquisition\n",
    "\n",
    "I selected C_fw075 after reviewing the runinfo.csv in the repository because it had 1 baseline and 1 follow-up cell free dna sample.\n",
    "\n",
    "Our pre-treatment sample is `SRR13973824` → sample label `C_fw075_baseline`\n",
    "Our post or follow-up treatment sample `SRR13973825` → sample label `C_fw075_followup`\n",
    "\n",
    "The raw data were obtained from the NCBI SRA project associated with the paper.  \n",
    "In practice, I pulled the data using `prefetch` and `fasterq-dump` (only run once on my machine). An example of the commands is:\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4f62429",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>LoadDate</th>\n",
       "      <th>spots</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots_with_mates</th>\n",
       "      <th>avgLength</th>\n",
       "      <th>size_MB</th>\n",
       "      <th>AssemblyName</th>\n",
       "      <th>download_path</th>\n",
       "      <th>...</th>\n",
       "      <th>Affection_Status</th>\n",
       "      <th>Analyte_Type</th>\n",
       "      <th>Histological_Type</th>\n",
       "      <th>Body_Site</th>\n",
       "      <th>CenterName</th>\n",
       "      <th>Submission</th>\n",
       "      <th>dbgap_study_accession</th>\n",
       "      <th>Consent</th>\n",
       "      <th>RunHash</th>\n",
       "      <th>ReadHash</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR13974039</td>\n",
       "      <td>2022-04-18 00:00:24</td>\n",
       "      <td>2021-03-16 08:45:24</td>\n",
       "      <td>2132268</td>\n",
       "      <td>615326977</td>\n",
       "      <td>2053223</td>\n",
       "      <td>288</td>\n",
       "      <td>350</td>\n",
       "      <td>GCA_000001405.13</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SEOUL NATIONAL UNIVERSITY</td>\n",
       "      <td>SRA1206760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>25A5EE99E44E4FBC0D6B95BE1221A156</td>\n",
       "      <td>5A979126CE22F541C18C1B3DB7394495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR13974041</td>\n",
       "      <td>2022-04-18 00:00:24</td>\n",
       "      <td>2021-03-16 08:45:29</td>\n",
       "      <td>2704436</td>\n",
       "      <td>789609381</td>\n",
       "      <td>2632994</td>\n",
       "      <td>291</td>\n",
       "      <td>420</td>\n",
       "      <td>GCA_000001405.13</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SEOUL NATIONAL UNIVERSITY</td>\n",
       "      <td>SRA1206760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>099ADF641FDB3EC474D711584108DBB1</td>\n",
       "      <td>86C1109112E35E3F10C96AFAB8216EC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR13974040</td>\n",
       "      <td>2022-04-18 00:00:24</td>\n",
       "      <td>2021-03-16 08:45:31</td>\n",
       "      <td>2263227</td>\n",
       "      <td>659576113</td>\n",
       "      <td>2218251</td>\n",
       "      <td>291</td>\n",
       "      <td>354</td>\n",
       "      <td>GCA_000001405.13</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SEOUL NATIONAL UNIVERSITY</td>\n",
       "      <td>SRA1206760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>C85AA55C37FB24B12E19C3B70DC7EC67</td>\n",
       "      <td>CBF81247DE6651D5F9633ACB7E61A8A4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR13974043</td>\n",
       "      <td>2022-04-18 00:00:24</td>\n",
       "      <td>2021-03-16 08:47:07</td>\n",
       "      <td>944806</td>\n",
       "      <td>265589338</td>\n",
       "      <td>900493</td>\n",
       "      <td>281</td>\n",
       "      <td>141</td>\n",
       "      <td>GCA_000001405.13</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SEOUL NATIONAL UNIVERSITY</td>\n",
       "      <td>SRA1206760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>B07E7B03747F8B6788FECE06D2052AB7</td>\n",
       "      <td>01C130C3118CF67695B55B2D77782834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR13974042</td>\n",
       "      <td>2022-04-18 00:00:24</td>\n",
       "      <td>2021-03-16 08:46:13</td>\n",
       "      <td>1374004</td>\n",
       "      <td>389796958</td>\n",
       "      <td>1360349</td>\n",
       "      <td>283</td>\n",
       "      <td>182</td>\n",
       "      <td>GCA_000001405.13</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SEOUL NATIONAL UNIVERSITY</td>\n",
       "      <td>SRA1206760</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>8BE2BA608B12D72AEFFB1939015AFFF7</td>\n",
       "      <td>23C18760023C61F4C910D0CDA575CE91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Run          ReleaseDate             LoadDate    spots      bases  \\\n",
       "0  SRR13974039  2022-04-18 00:00:24  2021-03-16 08:45:24  2132268  615326977   \n",
       "1  SRR13974041  2022-04-18 00:00:24  2021-03-16 08:45:29  2704436  789609381   \n",
       "2  SRR13974040  2022-04-18 00:00:24  2021-03-16 08:45:31  2263227  659576113   \n",
       "3  SRR13974043  2022-04-18 00:00:24  2021-03-16 08:47:07   944806  265589338   \n",
       "4  SRR13974042  2022-04-18 00:00:24  2021-03-16 08:46:13  1374004  389796958   \n",
       "\n",
       "   spots_with_mates  avgLength  size_MB      AssemblyName  \\\n",
       "0           2053223        288      350  GCA_000001405.13   \n",
       "1           2632994        291      420  GCA_000001405.13   \n",
       "2           2218251        291      354  GCA_000001405.13   \n",
       "3            900493        281      141  GCA_000001405.13   \n",
       "4           1360349        283      182  GCA_000001405.13   \n",
       "\n",
       "                                       download_path  ... Affection_Status  \\\n",
       "0  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "1  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "2  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "3  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "4  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "\n",
       "  Analyte_Type Histological_Type Body_Site                 CenterName  \\\n",
       "0          NaN               NaN       NaN  SEOUL NATIONAL UNIVERSITY   \n",
       "1          NaN               NaN       NaN  SEOUL NATIONAL UNIVERSITY   \n",
       "2          NaN               NaN       NaN  SEOUL NATIONAL UNIVERSITY   \n",
       "3          NaN               NaN       NaN  SEOUL NATIONAL UNIVERSITY   \n",
       "4          NaN               NaN       NaN  SEOUL NATIONAL UNIVERSITY   \n",
       "\n",
       "   Submission  dbgap_study_accession  Consent  \\\n",
       "0  SRA1206760                    NaN   public   \n",
       "1  SRA1206760                    NaN   public   \n",
       "2  SRA1206760                    NaN   public   \n",
       "3  SRA1206760                    NaN   public   \n",
       "4  SRA1206760                    NaN   public   \n",
       "\n",
       "                            RunHash                          ReadHash  \n",
       "0  25A5EE99E44E4FBC0D6B95BE1221A156  5A979126CE22F541C18C1B3DB7394495  \n",
       "1  099ADF641FDB3EC474D711584108DBB1  86C1109112E35E3F10C96AFAB8216EC2  \n",
       "2  C85AA55C37FB24B12E19C3B70DC7EC67  CBF81247DE6651D5F9633ACB7E61A8A4  \n",
       "3  B07E7B03747F8B6788FECE06D2052AB7  01C130C3118CF67695B55B2D77782834  \n",
       "4  8BE2BA608B12D72AEFFB1939015AFFF7  23C18760023C61F4C910D0CDA575CE91  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###First we need to inspect runinfo on NCBI repository to select a subject C_fw pre and post sample(follow-up)\n",
    "#I just downloaded this file when I was browsing and moved the .csv into the sra directory.\n",
    "\n",
    "runinfo_path = \"/home/mgpersonal/ctdna_kras_project/data/sra/PRJNA714799_runinfo.csv\"\n",
    "runinfo = pd.read_csv(runinfo_path)\n",
    "\n",
    "\n",
    "runinfo.head()\n",
    "\n",
    "###Had to manually inspect/confirm the SampleName column has the information we want.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "cb3f6501",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>SampleName</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>SRR13973824</td>\n",
       "      <td>C_fw075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>SRR13973825</td>\n",
       "      <td>C_fw075-2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Run SampleName\n",
       "208  SRR13973824    C_fw075\n",
       "209  SRR13973825  C_fw075-2"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###Identify the samples and the Run ID SRR\n",
    "samples = runinfo[\"SampleName\"].isin([\"C_fw075\", \"C_fw075-2\"])\n",
    "ctdna = runinfo.loc[samples, [\"Run\", \"SampleName\"]]\n",
    "ctdna\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "59f1d480",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "spots read      : 4,998,783\n",
      "reads read      : 9,997,566\n",
      "reads written   : 9,905,463\n",
      "spots read      : 5,508,730\n",
      "reads read      : 11,017,460\n",
      "reads written   : 10,946,198\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "project_root=\"/home/mgpersonal/ctdna_kras_project\"\n",
    "fastq_dir=\"${project_root}/data/fastq\"\n",
    "tmp_dir=\"${project_root}/data/sra_tmp\"\n",
    "\n",
    "mkdir -p \"${fastq_dir}\" \"${tmp_dir}\"\n",
    "\n",
    "for srr in SRR13973824 SRR13973825; do\n",
    "  fasterq-dump \\\n",
    "    --split-3 \\\n",
    "    --temp \"${tmp_dir}\" \\\n",
    "    -O \"${fastq_dir}\" \\\n",
    "    \"${srr}\"\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd90125d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  19626720 /home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973824_1.fastq\n",
      "  19626720 /home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973824_2.fastq\n",
      "  39253440 total\n",
      "  21749872 /home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973825_1.fastq\n",
      "  21749872 /home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973825_2.fastq\n",
      "  43499744 total\n"
     ]
    }
   ],
   "source": [
    "\n",
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "#need for shell\n",
    "project_root=\"/home/mgpersonal/ctdna_kras_project\"\n",
    "fastq_dir=\"${project_root}/data/fastq\"\n",
    "\n",
    "for srr in SRR13973824 SRR13973825; do\n",
    "  wc -l \"${fastq_dir}/${srr}_1.fastq\" \"${fastq_dir}/${srr}_2.fastq\"\n",
    "done\n",
    "\n",
    "##I was having an issue with different reads coming out of a 2 split so I made it a 3 split in the previous cell and now R1 and R2 work for pre and follow-up sample\n",
    "#this just verifies R1 and R2 are the same for bowtie2."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d2e37f4",
   "metadata": {},
   "source": [
    "# 4. QC - FASTQC assessment \n",
    "\n",
    "Sequenced quality was assessed for the C_fw075 subject's pre and follow-up forward and reverse fastq files. Read quality, GC content, and adapter content all appeared good. It didn't look like there was any striking adapter or issues with falloff at the tails.\n",
    "\n",
    "I decided not to perform any trimming because of the decent quality (> 30 fred). I will certainly reassess if I see anything concerning downstream. It looks like the authors of the paper do have a \"filtered\" file in their methods, but I couldn't find the exact method and I did not want to get too fixed on this for the puprose and scope of the assignment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "5dba84b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973824.fastq',\n",
       " '/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973824_1.fastq',\n",
       " '/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973824_2.fastq',\n",
       " '/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973825.fastq',\n",
       " '/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973825_1.fastq',\n",
       " '/home/mgpersonal/ctdna_kras_project/data/fastq/SRR13973825_2.fastq']"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###Check fastqs (not included) to make sure we have all pairs.\n",
    "##We can just keep the singletons in the .fastq\n",
    "\n",
    "fastqs = sorted(glob.glob(os.path.join(fastq_dir, \"*.fastq*\")))\n",
    "fastqs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8cd0bacc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  - SRR13973824.fastq\n",
      "  - SRR13973824_1.fastq\n",
      "  - SRR13973824_2.fastq\n",
      "  - SRR13973825.fastq\n",
      "  - SRR13973825_1.fastq\n",
      "  - SRR13973825_2.fastq\n",
      "null\n",
      "null\n",
      "Started analysis of SRR13973824.fastq\n",
      "null\n",
      "null\n",
      "null\n",
      "null\n",
      "Approx 5% complete for SRR13973824.fastq\n",
      "Approx 10% complete for SRR13973824.fastq\n",
      "Approx 15% complete for SRR13973824.fastq\n",
      "Approx 20% complete for SRR13973824.fastq\n",
      "Approx 25% complete for SRR13973824.fastq\n",
      "Approx 30% complete for SRR13973824.fastq\n",
      "Approx 35% complete for SRR13973824.fastq\n",
      "Approx 40% complete for SRR13973824.fastq\n",
      "Approx 45% complete for SRR13973824.fastq\n",
      "Approx 50% complete for SRR13973824.fastq\n",
      "Approx 55% complete for SRR13973824.fastq\n",
      "Approx 60% complete for SRR13973824.fastq\n",
      "Approx 65% complete for SRR13973824.fastq\n",
      "Approx 70% complete for SRR13973824.fastq\n",
      "Approx 75% complete for SRR13973824.fastq\n",
      "Approx 80% complete for SRR13973824.fastq\n",
      "Approx 85% complete for SRR13973824.fastq\n",
      "Approx 90% complete for SRR13973824.fastq\n",
      "Approx 95% complete for SRR13973824.fastq\n",
      "Analysis complete for SRR13973824.fastq\n",
      "Started analysis of SRR13973824_1.fastq\n",
      "Approx 5% complete for SRR13973824_1.fastq\n",
      "Approx 10% complete for SRR13973824_1.fastq\n",
      "Approx 15% complete for SRR13973824_1.fastq\n",
      "Approx 20% complete for SRR13973824_1.fastq\n",
      "Approx 25% complete for SRR13973824_1.fastq\n",
      "Approx 30% complete for SRR13973824_1.fastq\n",
      "Approx 35% complete for SRR13973824_1.fastq\n",
      "Approx 40% complete for SRR13973824_1.fastq\n",
      "Approx 45% complete for SRR13973824_1.fastq\n",
      "Approx 50% complete for SRR13973824_1.fastq\n",
      "Approx 55% complete for SRR13973824_1.fastq\n",
      "Approx 60% complete for SRR13973824_1.fastq\n",
      "Approx 65% complete for SRR13973824_1.fastq\n",
      "Approx 70% complete for SRR13973824_1.fastq\n",
      "Approx 75% complete for SRR13973824_1.fastq\n",
      "Approx 80% complete for SRR13973824_1.fastq\n",
      "Approx 85% complete for SRR13973824_1.fastq\n",
      "Approx 90% complete for SRR13973824_1.fastq\n",
      "Approx 95% complete for SRR13973824_1.fastq\n",
      "Analysis complete for SRR13973824_1.fastq\n",
      "Started analysis of SRR13973824_2.fastq\n",
      "Approx 5% complete for SRR13973824_2.fastq\n",
      "Approx 10% complete for SRR13973824_2.fastq\n",
      "Approx 15% complete for SRR13973824_2.fastq\n",
      "Approx 20% complete for SRR13973824_2.fastq\n",
      "Approx 25% complete for SRR13973824_2.fastq\n",
      "Approx 30% complete for SRR13973824_2.fastq\n",
      "Approx 35% complete for SRR13973824_2.fastq\n",
      "Approx 40% complete for SRR13973824_2.fastq\n",
      "Approx 45% complete for SRR13973824_2.fastq\n",
      "Approx 50% complete for SRR13973824_2.fastq\n",
      "Approx 55% complete for SRR13973824_2.fastq\n",
      "Approx 60% complete for SRR13973824_2.fastq\n",
      "Approx 65% complete for SRR13973824_2.fastq\n",
      "Approx 70% complete for SRR13973824_2.fastq\n",
      "Approx 75% complete for SRR13973824_2.fastq\n",
      "Approx 80% complete for SRR13973824_2.fastq\n",
      "Approx 85% complete for SRR13973824_2.fastq\n",
      "Approx 90% complete for SRR13973824_2.fastq\n",
      "Approx 95% complete for SRR13973824_2.fastq\n",
      "Analysis complete for SRR13973824_2.fastq\n",
      "Started analysis of SRR13973825.fastq\n",
      "Approx 5% complete for SRR13973825.fastq\n",
      "Approx 10% complete for SRR13973825.fastq\n",
      "Approx 15% complete for SRR13973825.fastq\n",
      "Approx 20% complete for SRR13973825.fastq\n",
      "Approx 25% complete for SRR13973825.fastq\n",
      "Approx 30% complete for SRR13973825.fastq\n",
      "Approx 35% complete for SRR13973825.fastq\n",
      "Approx 40% complete for SRR13973825.fastq\n",
      "Approx 45% complete for SRR13973825.fastq\n",
      "Approx 50% complete for SRR13973825.fastq\n",
      "Approx 55% complete for SRR13973825.fastq\n",
      "Approx 60% complete for SRR13973825.fastq\n",
      "Approx 65% complete for SRR13973825.fastq\n",
      "Approx 70% complete for SRR13973825.fastq\n",
      "Approx 75% complete for SRR13973825.fastq\n",
      "Approx 80% complete for SRR13973825.fastq\n",
      "Approx 85% complete for SRR13973825.fastq\n",
      "Approx 90% complete for SRR13973825.fastq\n",
      "Approx 95% complete for SRR13973825.fastq\n",
      "Analysis complete for SRR13973825.fastq\n",
      "Started analysis of SRR13973825_1.fastq\n",
      "Approx 5% complete for SRR13973825_1.fastq\n",
      "Approx 10% complete for SRR13973825_1.fastq\n",
      "Approx 15% complete for SRR13973825_1.fastq\n",
      "Approx 20% complete for SRR13973825_1.fastq\n",
      "Approx 25% complete for SRR13973825_1.fastq\n",
      "Approx 30% complete for SRR13973825_1.fastq\n",
      "Approx 35% complete for SRR13973825_1.fastq\n",
      "Approx 40% complete for SRR13973825_1.fastq\n",
      "Approx 45% complete for SRR13973825_1.fastq\n",
      "Approx 50% complete for SRR13973825_1.fastq\n",
      "Approx 55% complete for SRR13973825_1.fastq\n",
      "Approx 60% complete for SRR13973825_1.fastq\n",
      "Approx 65% complete for SRR13973825_1.fastq\n",
      "Approx 70% complete for SRR13973825_1.fastq\n",
      "Approx 75% complete for SRR13973825_1.fastq\n",
      "Approx 80% complete for SRR13973825_1.fastq\n",
      "Approx 85% complete for SRR13973825_1.fastq\n",
      "Approx 90% complete for SRR13973825_1.fastq\n",
      "Approx 95% complete for SRR13973825_1.fastq\n",
      "Analysis complete for SRR13973825_1.fastq\n",
      "Started analysis of SRR13973825_2.fastq\n",
      "Approx 5% complete for SRR13973825_2.fastq\n",
      "Approx 10% complete for SRR13973825_2.fastq\n",
      "Approx 15% complete for SRR13973825_2.fastq\n",
      "Approx 20% complete for SRR13973825_2.fastq\n",
      "Approx 25% complete for SRR13973825_2.fastq\n",
      "Approx 30% complete for SRR13973825_2.fastq\n",
      "Approx 35% complete for SRR13973825_2.fastq\n",
      "Approx 40% complete for SRR13973825_2.fastq\n",
      "Approx 45% complete for SRR13973825_2.fastq\n",
      "Approx 50% complete for SRR13973825_2.fastq\n",
      "Approx 55% complete for SRR13973825_2.fastq\n",
      "Approx 60% complete for SRR13973825_2.fastq\n",
      "Approx 65% complete for SRR13973825_2.fastq\n",
      "Approx 70% complete for SRR13973825_2.fastq\n",
      "Approx 75% complete for SRR13973825_2.fastq\n",
      "Approx 80% complete for SRR13973825_2.fastq\n",
      "Approx 85% complete for SRR13973825_2.fastq\n",
      "Approx 90% complete for SRR13973825_2.fastq\n",
      "Approx 95% complete for SRR13973825_2.fastq\n",
      "Analysis complete for SRR13973825_2.fastq\n"
     ]
    }
   ],
   "source": [
    "###Running fASTQC for Read Quality Assessment\n",
    "#Asess raw read quality for the baseline and follow-up ctDNA fastq files\n",
    "\n",
    "# Find all FASTQ files in fastq_dir\n",
    "fastqs = sorted(glob.glob(os.path.join(fastq_dir, \"*.fastq\")))\n",
    "\n",
    "for f in fastqs:\n",
    "    print(\"  -\", os.path.basename(f))\n",
    "\n",
    "# Need to add a space for FASTQC command to work\n",
    "fastqs_str = \" \".join(fastqs)\n",
    "\n",
    "# Run FastQC\n",
    "!fastqc -o {qc_dir} {fastqs_str}\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e95a3b97",
   "metadata": {},
   "source": [
    "5. Analysis\n",
    "\n",
    "Paired-reads of the baseline and follow-up samples are aligned to the GRCh38 reference genome using bowtie2. The sorted.bams are indexed and we perform variant calling with VarScan mpileup2cn restricted to the KRAS region chr12:25245300–25245400. The VCF parameters were very flexible to try to detect any small fraction mutation that might appear, or even a de-novo alteration. The VCF files for baseline and follow-up samples were annotated with SnipEff (VEP and ClinVar would be nice, but they were not working well in my environment. COSMIC is unfortanately not available for personal non-affiliated use) There was evidence of the clinically meaningful mutation KRAS G12D present in the baseline sample but not the follow-up sample suggesting the metastatic CRC is responding to cetuximab treatment.\n",
    "\n",
    "Below are the key finding and commentary:\n",
    "\n",
    "1. Baseline Sample: Clear Detection of Oncogenic KRAS G12D\n",
    "\n",
    "In the baseline sample (C_fw075_baseline), we observe a well-supported KRAS G12D mutation (cDNA change 35G>A; Protein change Gly12Asp):\n",
    "\n",
    "Position: chr12:25245350\n",
    "Reference / Alt: C>T (GRCh38)\n",
    "Depth: ~1,300× total coverage\n",
    "ALT allele count: ~110 reads\n",
    "VAF: ~8.3%\n",
    "Annotation: missense_variant, impact: MODERATE\n",
    "\n",
    "KRAS G12D is a well-known hotspot mutation and a common driver of several cancers including CRC. In the baseline sample it is present at  about 8% VAF. Since this is from circulating DNA it likely would have a smaller VAF. Importantly, the mutation is supported by high depth and can be considered a robust call.\n",
    "\n",
    "2. Follow-up Sample: Absence of KRAS G12D and No High-Confidence Driver Mutations\n",
    "\n",
    "In the follow-up sample (C_fw075_followup), no G12D mutation is detected, even at low allele frequencies. Instead, the only KRAS-related variant observed is:\n",
    "\n",
    "Position: chr12:25250747\n",
    "Change: G>A\n",
    "Effect: splice_region_variant & intron_variant\n",
    "Depth: 46×\n",
    "ALT allele count: 2 \n",
    "VAF: 4%\n",
    "Annotation impact: LOW\n",
    "\n",
    "Even though this variant has an LOW annotation impact from SnpEFF, there is not too much known about it. It is also at a very low depth and VAF. For these 2 reasons it can be considered a weak call, if not background. The absence of the KRAS G12D alteration is likely due to cetuximab treatment response. It can be a bit reassuring to the patient that the levels of that oncogenic driver are not detectable post treatment.\n",
    "\n",
    "It is also good to see no new robust KRAS alteration appear like G12C, A, R, etc. even though this is not the usual resistance mechanism to cetuximab. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "4495df7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973824_1_fastqc.html',\n",
       " '/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973824_2_fastqc.html',\n",
       " '/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973824_fastqc.html',\n",
       " '/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973825_1_fastqc.html',\n",
       " '/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973825_2_fastqc.html',\n",
       " '/home/mgpersonal/ctdna_kras_project/results/qc/SRR13973825_fastqc.html']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "###Review the FASTQC .hmtl files\n",
    "#It looks like the sequences are pretty good quality (>30) in all the files. Sure the adapters and tails fall off a bit, but since I am less familiar with this exact targetted sequencing, I am going to avoid trimming unless something looks concerning down the line.\n",
    "qc_dir = \"/home/mgpersonal/ctdna_kras_project/results/qc\"\n",
    "sorted(glob.glob(os.path.join(qc_dir, \"*fastqc.html\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd993edf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning SRR13973824 (C_fw075_baseline)...\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "#need to def these again for this bash shell\n",
    "project_root=\"/home/mgpersonal/ctdna_kras_project\"\n",
    "fastq_dir=\"${project_root}/data/fastq\"\n",
    "ref_dir=\"${project_root}/data/ref\"\n",
    "bam_dir=\"${project_root}/results/bam\"\n",
    "\n",
    "ref_prefix=\"${ref_dir}/GRCh38_primary\"\n",
    "\n",
    "\n",
    "sra_id=\"SRR13973824\"\n",
    "label=\"C_fw075_baseline\"\n",
    "\n",
    "R1=\"${fastq_dir}/${sra_id}_1.fastq\"\n",
    "R2=\"${fastq_dir}/${sra_id}_2.fastq\"\n",
    "\n",
    "echo \"Aligning ${sra_id} (${label})...\"\n",
    "bowtie2 -x \"${ref_prefix}\" \\\n",
    "  -1 \"${R1}\" -2 \"${R2}\" \\\n",
    "  -p 8 \\\n",
    "  2> \"${bam_dir}/${label}.bowtie2.log\" \\\n",
    "| samtools sort -o \"${bam_dir}/${label}.sorted.bam\"\n",
    "\n",
    "#.sorted.bam created for pre sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52d1f675",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning SRR13973825 (C_fw075_followup)...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "#need to def these again for this bash shell\n",
    "project_root=\"/home/mgpersonal/ctdna_kras_project\"\n",
    "fastq_dir=\"${project_root}/data/fastq\"\n",
    "ref_dir=\"${project_root}/data/ref\"\n",
    "bam_dir=\"${project_root}/results/bam\"\n",
    "\n",
    "ref_prefix=\"${ref_dir}/GRCh38_primary\"\n",
    "\n",
    "#change these for follow-up\n",
    "sra_id=\"SRR13973825\"\n",
    "label=\"C_fw075_followup\"\n",
    "\n",
    "R1=\"${fastq_dir}/${sra_id}_1.fastq\"\n",
    "R2=\"${fastq_dir}/${sra_id}_2.fastq\"\n",
    "\n",
    "echo \"Aligning ${sra_id} (${label})...\"\n",
    "bowtie2 -x \"${ref_prefix}\" \\\n",
    "  -1 \"${R1}\" -2 \"${R2}\" \\\n",
    "  -p 8 \\\n",
    "  2> \"${bam_dir}/${label}.bowtie2.log\" \\\n",
    "| samtools sort -o \"${bam_dir}/${label}.sorted.bam\"\n",
    "\n",
    "#.sorted.bam created for follow-up sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c4b0501",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Indexing C_fw075_baseline.sorted.bam\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Indexing C_fw075_followup.sorted.bam\n",
      "Done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "bam_dir=\"/home/mgpersonal/ctdna_kras_project/results/bam\"\n",
    "cd \"${bam_dir}\"\n",
    "\n",
    "for bam in *.sorted.bam; do\n",
    "    echo \"Indexing $bam\"\n",
    "    samtools index \"$bam\"\n",
    "done\n",
    "\n",
    "echo \"Done.\"\n",
    "\n",
    "\n",
    "#we are just indexing our .sorted.bams here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23b634cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BAMs:\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 846M Nov 15 14:38 /home/mgpersonal/ctdna_kras_project/results/bam/C_fw075_baseline.sorted.bam\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 900M Nov 15 14:51 /home/mgpersonal/ctdna_kras_project/results/bam/C_fw075_followup.sorted.bam\n",
      "\n",
      "mpileup + VarScan for C_fw075_baseline\n",
      "  BAM: /home/mgpersonal/ctdna_kras_project/results/bam/C_fw075_baseline.sorted.bam\n",
      "  Region: 12:25204000-25253000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[mpileup] 1 samples in 1 input files\n",
      "/home/mgpersonal/miniconda3/envs/ngs/bin/varscan: line 6: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory\n",
      "Min coverage:\t10\n",
      "Min reads2:\t2\n",
      "Min var freq:\t0.01\n",
      "Min avg qual:\t25\n",
      "P-value thresh:\t0.99\n",
      "Reading input from /home/mgpersonal/ctdna_kras_project/results/vcf/C_fw075_baseline.kras.mpileup\n",
      "9037 bases in pileup file\n",
      "28 variant positions (21 SNP, 7 indel)\n",
      "5 were failed by the strand-filter\n",
      "28 variant positions reported (21 SNP, 7 indel)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  vcf created: /home/mgpersonal/ctdna_kras_project/results/vcf/C_fw075_baseline.kras.varscan.vcf\n",
      "\n",
      "mpileup + VarScan for C_fw075_followup\n",
      "  BAM: /home/mgpersonal/ctdna_kras_project/results/bam/C_fw075_followup.sorted.bam\n",
      "  Region: 12:25204000-25253000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[mpileup] 1 samples in 1 input files\n",
      "/home/mgpersonal/miniconda3/envs/ngs/bin/varscan: line 6: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory\n",
      "Min coverage:\t10\n",
      "Min reads2:\t2\n",
      "Min var freq:\t0.01\n",
      "Min avg qual:\t25\n",
      "P-value thresh:\t0.99\n",
      "Reading input from /home/mgpersonal/ctdna_kras_project/results/vcf/C_fw075_followup.kras.mpileup\n",
      "8233 bases in pileup file\n",
      "20 variant positions (13 SNP, 7 indel)\n",
      "2 were failed by the strand-filter\n",
      "20 variant positions reported (13 SNP, 7 indel)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  vcf created: /home/mgpersonal/ctdna_kras_project/results/vcf/C_fw075_followup.kras.varscan.vcf\n",
      "\n",
      "Done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "\n",
    "project_root=\"/home/mgpersonal/ctdna_kras_project\"\n",
    "ref_dir=\"${project_root}/data/ref\"\n",
    "bam_dir=\"${project_root}/results/bam\"\n",
    "vcf_dir=\"${project_root}/results/vcf\"\n",
    "\n",
    "ref_fasta=\"${ref_dir}/Homo_sapiens.GRCh38.dna.primary_assembly.fa\"\n",
    "# looked up GRCh38 KRAS coordinates on Ensembl 12: 25,205,246-25,250,936 reverse strand. Padded the region slightly.\n",
    "region=\"12:25204000-25253000\"\n",
    "\n",
    "# Let's confirm our sorted sorted.bams\n",
    "echo \"BAMs:\"\n",
    "ls -lh \"${bam_dir}\"/*.sorted.bam || true\n",
    "\n",
    "# loop over pre and follow-up sample creating mpileup straight to VCF file of KRAS region\n",
    "for label in C_fw075_baseline C_fw075_followup; do\n",
    "    bam=\"${bam_dir}/${label}.sorted.bam\"\n",
    "    mpileup=\"${vcf_dir}/${label}.kras.mpileup\"\n",
    "    vcf=\"${vcf_dir}/${label}.kras.varscan.vcf\"\n",
    "\n",
    "    echo\n",
    "    echo \"mpileup + VarScan for ${label}\"\n",
    "    echo \"  BAM: ${bam}\"\n",
    "    echo \"  Region: ${region}\"\n",
    "\n",
    "    # mpileup restricted to KRAS region\n",
    "    samtools mpileup -f \"${ref_fasta}\" -r \"${region}\" \"${bam}\" > \"${mpileup}\"\n",
    "\n",
    "    # VarScan consensus \n",
    "    #\n",
    "    varscan mpileup2cns \"${mpileup}\" \\\n",
    "        --min-coverage 10 \\\n",
    "        --min-avg-qual 25 \\\n",
    "        --min-var-freq 0.01 \\\n",
    "        --p-value 0.99 \\\n",
    "        --output-vcf 1 > \"${vcf}\"\n",
    "\n",
    "    echo \"  vcf created: ${vcf}\"\n",
    "done\n",
    "\n",
    "echo\n",
    "echo \"Done.\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "4c2bf1f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "00:00:00 SnpEff version SnpEff 5.3a (build 2025-09-02 10:24), by Pablo Cingolani\n",
      "00:00:00 Command: 'ann'\n",
      "00:00:00 Reading configuration file 'snpEff.config'. Genome: 'GRCh38.99'\n",
      "00:00:00 Looking for config file: '/home/mgpersonal/ctdna_kras_project/results/vcf/snpEff.config'\n",
      "00:00:00 Reading config file: /home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/snpEff.config\n",
      "00:00:00 done\n",
      "00:00:00 Reading database for genome version 'GRCh38.99' from file '/home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/./data/GRCh38.99/snpEffectPredictor.bin' (this might take a while)\n",
      "00:00:13 done\n",
      "00:00:13 Loading Motifs and PWMs\n",
      "00:00:13 Building interval forest\n",
      "00:00:17 done.\n",
      "00:00:17 Genome stats :\n",
      "#-----------------------------------------------\n",
      "# Genome name                : 'Homo_sapiens'\n",
      "# Genome version             : 'GRCh38.99'\n",
      "# Genome ID                  : 'GRCh38.99[0]'\n",
      "# Has protein coding info    : true\n",
      "# Has Tr. Support Level info : true\n",
      "# Genes                      : 60676\n",
      "# Protein coding genes       : 20135\n",
      "#-----------------------------------------------\n",
      "# Transcripts                : 227818\n",
      "# Avg. transcripts per gene  : 3.75\n",
      "# TSL transcripts            : 169199\n",
      "#-----------------------------------------------\n",
      "# Checked transcripts        : \n",
      "#               AA sequences : 100576 ( 119.03% )\n",
      "#              DNA sequences : 172288 ( 75.63% )\n",
      "#-----------------------------------------------\n",
      "# Protein coding transcripts : 84494\n",
      "#              Length errors :  13469 ( 15.94% )\n",
      "#  STOP codons in CDS errors :     34 ( 0.04% )\n",
      "#         START codon errors :   8149 ( 9.64% )\n",
      "#        STOP codon warnings :   7264 ( 8.60% )\n",
      "#              UTR sequences :  81637 ( 35.83% )\n",
      "#               Total Errors :  20119 ( 23.81% )\n",
      "#-----------------------------------------------\n",
      "# Cds                        : 764354\n",
      "# Exons                      : 1376499\n",
      "# Exons with sequence        : 1376499\n",
      "# Exons without sequence     : 0\n",
      "# Avg. exons per transcript  : 6.04\n",
      "# WARNING!                   : Mitochondrion chromosome 'MT' does not have a mitochondrion codon table (codon table = 'Standard'). You should update the config file.\n",
      "#-----------------------------------------------\n",
      "# Number of chromosomes      : 639\n",
      "# Chromosomes                : Format 'chromo_name size codon_table'\n",
      "#\t\t'HSCHR1_2_CTG3'\t248975002\tStandard\n",
      "#\t\t'HSCHR1_1_CTG31'\t248973653\tStandard\n",
      "#\t\t'HSCHR1_2_CTG31'\t248971826\tStandard\n",
      "#\t\t'HSCHR1_1_CTG3'\t248968406\tStandard\n",
      "#\t\t'HSCHR1_3_CTG31'\t248967909\tStandard\n",
      "#\t\t'HSCHR1_9_CTG3'\t248965380\tStandard\n",
      "#\t\t'HSCHR1_4_CTG3'\t248963886\tStandard\n",
      "#\t\t'HSCHR1_2_CTG32_1'\t248962488\tStandard\n",
      "#\t\t'HSCHR1_8_CTG3'\t248962151\tStandard\n",
      "#\t\t'HSCHR1_3_CTG3'\t248961700\tStandard\n",
      "#\t\t'HSCHR1_1_CTG11'\t248961539\tStandard\n",
      "#\t\t'HSCHR1_3_CTG32_1'\t248960761\tStandard\n",
      "#\t\t'HSCHR1_5_CTG32_1'\t248960213\tStandard\n",
      "#\t\t'HSCHR1_1_CTG32_1'\t248960143\tStandard\n",
      "#\t\t'HG1832_PATCH'\t248959779\tStandard\n",
      "#\t\t'HG2095_PATCH'\t248959131\tStandard\n",
      "#\t\t'HSCHR1_ALT2_1_CTG32_1'\t248958866\tStandard\n",
      "#\t\t'HSCHR1_4_CTG32_1'\t248958766\tStandard\n",
      "#\t\t'HG2104_PATCH'\t248958259\tStandard\n",
      "#\t\t'HG986_PATCH'\t248956639\tStandard\n",
      "#\t\t'1'\t248956422\tStandard\n",
      "#\t\t'HG2058_PATCH'\t248956422\tStandard\n",
      "#\t\t'HSCHR1_6_CTG3'\t248955842\tStandard\n",
      "#\t\t'HG460_PATCH'\t248955838\tStandard\n",
      "#\t\t'HSCHR1_4_CTG31'\t248955631\tStandard\n",
      "#\t\t'HG2002_PATCH'\t248942227\tStandard\n",
      "#\t\t'HG1342_HG2282_PATCH'\t248929249\tStandard\n",
      "#\t\t'HSCHR1_5_CTG3'\t248926662\tStandard\n",
      "#\t\t'HG2290_PATCH'\t242235269\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7'\t242223512\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7_2'\t242210518\tStandard\n",
      "#\t\t'HSCHR2_7_CTG7_2'\t242202531\tStandard\n",
      "#\t\t'HSCHR2_5_CTG7_2'\t242199685\tStandard\n",
      "#\t\t'HSCHR2_8_CTG7_2'\t242199670\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7_2'\t242199609\tStandard\n",
      "#\t\t'HSCHR2_1_CTG5'\t242199562\tStandard\n",
      "#\t\t'HSCHR2_1_CTG15'\t242198904\tStandard\n",
      "#\t\t'HSCHR2_2_CTG15'\t242198544\tStandard\n",
      "#\t\t'HSCHR2_2_CTG1'\t242198346\tStandard\n",
      "#\t\t'HSCHR2_6_CTG7_2'\t242198149\tStandard\n",
      "#\t\t'HSCHR2_4_CTG7_2'\t242196589\tStandard\n",
      "#\t\t'HSCHR2_3_CTG1'\t242195626\tStandard\n",
      "#\t\t'HG1384_PATCH'\t242195459\tStandard\n",
      "#\t\t'HSCHR2_1_CTG1'\t242194896\tStandard\n",
      "#\t\t'HSCHR2_3_CTG7_2'\t242194557\tStandard\n",
      "#\t\t'HSCHR2_3_CTG15'\t242194292\tStandard\n",
      "#\t\t'HG721_PATCH'\t242194216\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7'\t242193540\tStandard\n",
      "#\t\t'HG2233_PATCH'\t242193535\tStandard\n",
      "#\t\t'2'\t242193529\tStandard\n",
      "#\t\t'HG2232_PATCH'\t242193529\tStandard\n",
      "#\t\t'HSCHR2_4_CTG1'\t242191076\tStandard\n",
      "#\t\t'HSCHR3_4_CTG1'\t198310749\tStandard\n",
      "#\t\t'HSCHR3_1_CTG2_1'\t198306097\tStandard\n",
      "#\t\t'HSCHR3_5_CTG1'\t198305844\tStandard\n",
      "#\t\t'HSCHR3_6_CTG2_1'\t198305007\tStandard\n",
      "#\t\t'HSCHR3_1_CTG3'\t198302882\tStandard\n",
      "#\t\t'HSCHR3_9_CTG2_1'\t198302489\tStandard\n",
      "#\t\t'HSCHR3_8_CTG2_1'\t198302280\tStandard\n",
      "#\t\t'HSCHR3_4_CTG2_1'\t198301383\tStandard\n",
      "#\t\t'HSCHR3_7_CTG2_1'\t198301024\tStandard\n",
      "#\t\t'HSCHR3_5_CTG2_1'\t198300594\tStandard\n",
      "#\t\t'HSCHR3_3_CTG2_1'\t198300421\tStandard\n",
      "#\t\t'HSCHR3_3_CTG1'\t198300110\tStandard\n",
      "#\t\t'HG2133_PATCH'\t198299611\tStandard\n",
      "#\t\t'HSCHR3_2_CTG2_1'\t198298736\tStandard\n",
      "#\t\t'HSCHR3_1_CTG1'\t198297480\tStandard\n",
      "#\t\t'HSCHR3_2_CTG3'\t198296961\tStandard\n",
      "#\t\t'HSCHR3_4_CTG3'\t198296867\tStandard\n",
      "#\t\t'HG2022_PATCH'\t198295941\tStandard\n",
      "#\t\t'HSCHR3_8_CTG3'\t198295934\tStandard\n",
      "#\t\t'HG126_PATCH'\t198295908\tStandard\n",
      "#\t\t'HG2235_PATCH'\t198295655\tStandard\n",
      "#\t\t'3'\t198295559\tStandard\n",
      "#\t\t'HG2066_PATCH'\t198295555\tStandard\n",
      "#\t\t'HG2236_PATCH'\t198295537\tStandard\n",
      "#\t\t'HSCHR3_6_CTG3'\t198294921\tStandard\n",
      "#\t\t'HSCHR3_7_CTG3'\t198294497\tStandard\n",
      "#\t\t'HSCHR3_5_CTG3'\t198293785\tStandard\n",
      "#\t\t'HSCHR3_3_CTG3'\t198293223\tStandard\n",
      "#\t\t'HSCHR3_9_CTG3'\t198291952\tStandard\n",
      "#\t\t'HG2237_PATCH'\t198290329\tStandard\n",
      "#\t\t'HG2525_PATCH'\t190293418\tStandard\n",
      "#\t\t'HSCHR4_11_CTG12'\t190233769\tStandard\n",
      "#\t\t'HSCHR4_1_CTG12'\t190227204\tStandard\n",
      "#\t\t'HSCHR4_7_CTG12'\t190225141\tStandard\n",
      "#\t\t'HSCHR4_1_CTG6'\t190224838\tStandard\n",
      "#\t\t'HSCHR4_1_CTG4'\t190221757\tStandard\n",
      "#\t\t'HSCHR4_12_CTG12'\t190221648\tStandard\n",
      "#\t\t'HSCHR4_4_CTG12'\t190220704\tStandard\n",
      "#\t\t'HSCHR4_5_CTG12'\t190219562\tStandard\n",
      "#\t\t'HSCHR4_9_CTG12'\t190219005\tStandard\n",
      "#\t\t'HSCHR4_1_CTG8_1'\t190218838\tStandard\n",
      "#\t\t'HSCHR4_6_CTG12'\t190218533\tStandard\n",
      "#\t\t'HSCHR4_8_CTG12'\t190218480\tStandard\n",
      "#\t\t'HSCHR4_3_CTG12'\t190218028\tStandard\n",
      "#\t\t'HSCHR4_2_CTG4'\t190217898\tStandard\n",
      "#\t\t'4'\t190214555\tStandard\n",
      "#\t\t'HG699_PATCH'\t190211220\tStandard\n",
      "#\t\t'HG1296_PATCH'\t190208697\tStandard\n",
      "#\t\t'HSCHR4_2_CTG12'\t190206797\tStandard\n",
      "#\t\t'HG1299_PATCH'\t190201596\tStandard\n",
      "#\t\t'HG1298_PATCH'\t190196285\tStandard\n",
      "#\t\t'HG705_PATCH'\t190172551\tStandard\n",
      "#\t\t'HG2023_PATCH'\t190147504\tStandard\n",
      "#\t\t'HSCHR4_1_CTG9'\t190096932\tStandard\n",
      "#\t\t'HSCHR5_8_CTG1'\t181668817\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1'\t181558402\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1_1'\t181554865\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1'\t181547139\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1_1'\t181546281\tStandard\n",
      "#\t\t'HSCHR5_7_CTG1'\t181544732\tStandard\n",
      "#\t\t'HSCHR5_1_CTG5'\t181543736\tStandard\n",
      "#\t\t'HSCHR5_9_CTG1'\t181543440\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1'\t181542495\tStandard\n",
      "#\t\t'HSCHR5_2_CTG5'\t181542206\tStandard\n",
      "#\t\t'HSCHR5_6_CTG1'\t181540245\tStandard\n",
      "#\t\t'HSCHR5_5_CTG1'\t181539933\tStandard\n",
      "#\t\t'HG30_PATCH'\t181539112\tStandard\n",
      "#\t\t'HSCHR5_3_CTG5'\t181538569\tStandard\n",
      "#\t\t'5'\t181538259\tStandard\n",
      "#\t\t'HG1395_PATCH'\t181537525\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1'\t181537434\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1_1'\t180863570\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1_1'\t180753562\tStandard\n",
      "#\t\t'HSCHR6_MHC_MCF_CTG1'\t170976334\tStandard\n",
      "#\t\t'HSCHR6_MHC_MANN_CTG1'\t170959829\tStandard\n",
      "#\t\t'HSCHR6_MHC_SSTO_CTG1'\t170946136\tStandard\n",
      "#\t\t'HSCHR6_MHC_APD_CTG1'\t170845044\tStandard\n",
      "#\t\t'HSCHR6_1_CTG3'\t170836231\tStandard\n",
      "#\t\t'HSCHR6_1_CTG2'\t170833811\tStandard\n",
      "#\t\t'HSCHR6_1_CTG6'\t170818797\tStandard\n",
      "#\t\t'HG2072_PATCH'\t170816182\tStandard\n",
      "#\t\t'HSCHR6_1_CTG9'\t170815583\tStandard\n",
      "#\t\t'HSCHR6_1_CTG5'\t170815302\tStandard\n",
      "#\t\t'HSCHR6_8_CTG1'\t170813052\tStandard\n",
      "#\t\t'HSCHR6_1_CTG7'\t170812567\tStandard\n",
      "#\t\t'HG2128_PATCH'\t170811684\tStandard\n",
      "#\t\t'HSCHR6_1_CTG10'\t170810830\tStandard\n",
      "#\t\t'HSCHR6_1_CTG8'\t170810115\tStandard\n",
      "#\t\t'HSCHR6_1_CTG4'\t170808607\tStandard\n",
      "#\t\t'HG2121_PATCH'\t170808322\tStandard\n",
      "#\t\t'6'\t170805979\tStandard\n",
      "#\t\t'HG2057_PATCH'\t170805979\tStandard\n",
      "#\t\t'HSCHR6_MHC_DBB_CTG1'\t170783898\tStandard\n",
      "#\t\t'HG1651_PATCH'\t170779650\tStandard\n",
      "#\t\t'HG563_PATCH'\t170759097\tStandard\n",
      "#\t\t'HSCHR6_MHC_QBL_CTG1'\t170734801\tStandard\n",
      "#\t\t'HSCHR6_MHC_COX_CTG1'\t170727598\tStandard\n",
      "#\t\t'HSCHR7_3_CTG4_4'\t159441130\tStandard\n",
      "#\t\t'HSCHR7_2_CTG6'\t159407160\tStandard\n",
      "#\t\t'HG1309_PATCH'\t159365091\tStandard\n",
      "#\t\t'HSCHR7_3_CTG1'\t159352526\tStandard\n",
      "#\t\t'HSCHR7_1_CTG6'\t159351981\tStandard\n",
      "#\t\t'HG2088_PATCH'\t159351667\tStandard\n",
      "#\t\t'HSCHR7_2_CTG7'\t159349352\tStandard\n",
      "#\t\t'HSCHR7_1_CTG1'\t159349278\tStandard\n",
      "#\t\t'HSCHR7_1_CTG4_4'\t159348604\tStandard\n",
      "#\t\t'HSCHR7_2_CTG1'\t159348294\tStandard\n",
      "#\t\t'HSCHR7_1_CTG7'\t159346281\tStandard\n",
      "#\t\t'HG2266_PATCH'\t159346106\tStandard\n",
      "#\t\t'7'\t159345973\tStandard\n",
      "#\t\t'HG2239_PATCH'\t159345319\tStandard\n",
      "#\t\t'HSCHR7_3_CTG6'\t159344441\tStandard\n",
      "#\t\t'HSCHR7_2_CTG4_4'\t159333789\tStandard\n",
      "#\t\t'HG708_PATCH'\t159270839\tStandard\n",
      "#\t\t'HSCHRX_2_CTG12'\t156053172\tStandard\n",
      "#\t\t'HSCHRX_3_CTG7'\t156050387\tStandard\n",
      "#\t\t'HSCHRX_1_CTG3'\t156043585\tStandard\n",
      "#\t\t'X'\t156040895\tStandard\n",
      "#\t\t'HSCHRX_2_CTG3'\t156039875\tStandard\n",
      "#\t\t'HG1507_PATCH'\t155996656\tStandard\n",
      "#\t\t'HG1506_PATCH'\t155994589\tStandard\n",
      "#\t\t'HG1509_PATCH'\t155994530\tStandard\n",
      "#\t\t'HG439_PATCH'\t155960230\tStandard\n",
      "#\t\t'HG1466_PATCH'\t155848651\tStandard\n",
      "#\t\t'HSCHR8_1_CTG6'\t145150400\tStandard\n",
      "#\t\t'HSCHR8_7_CTG7'\t145147686\tStandard\n",
      "#\t\t'HSCHR8_5_CTG7'\t145144538\tStandard\n",
      "#\t\t'HG2067_PATCH'\t145144434\tStandard\n",
      "#\t\t'HSCHR8_4_CTG7'\t145143933\tStandard\n",
      "#\t\t'HSCHR8_1_CTG7'\t145142033\tStandard\n",
      "#\t\t'HSCHR8_2_CTG7'\t145141868\tStandard\n",
      "#\t\t'HSCHR8_6_CTG7'\t145141837\tStandard\n",
      "#\t\t'HSCHR8_2_CTG1'\t145141567\tStandard\n",
      "#\t\t'HSCHR8_4_CTG1'\t145141010\tStandard\n",
      "#\t\t'HG2068_PATCH'\t145140411\tStandard\n",
      "#\t\t'8'\t145138636\tStandard\n",
      "#\t\t'HG2419_PATCH'\t145138635\tStandard\n",
      "#\t\t'HSCHR8_7_CTG1'\t145138198\tStandard\n",
      "#\t\t'HSCHR8_5_CTG1'\t145137619\tStandard\n",
      "#\t\t'HSCHR8_6_CTG1'\t145136257\tStandard\n",
      "#\t\t'HSCHR8_3_CTG1'\t145130612\tStandard\n",
      "#\t\t'HSCHR8_3_CTG7'\t145125643\tStandard\n",
      "#\t\t'HSCHR8_1_CTG1'\t145124952\tStandard\n",
      "#\t\t'HSCHR8_8_CTG1'\t145123326\tStandard\n",
      "#\t\t'HSCHR8_9_CTG1'\t144983589\tStandard\n",
      "#\t\t'HG76_PATCH'\t144938989\tStandard\n",
      "#\t\t'HSCHR9_1_CTG1'\t138409529\tStandard\n",
      "#\t\t'HSCHR9_1_CTG4'\t138405815\tStandard\n",
      "#\t\t'HSCHR9_1_CTG5'\t138404305\tStandard\n",
      "#\t\t'HSCHR9_1_CTG2'\t138402913\tStandard\n",
      "#\t\t'HSCHR9_1_CTG3'\t138400797\tStandard\n",
      "#\t\t'HSCHR9_1_CTG6'\t138399532\tStandard\n",
      "#\t\t'HSCHR9_1_CTG7'\t138399111\tStandard\n",
      "#\t\t'HG613_PATCH'\t138394874\tStandard\n",
      "#\t\t'HG2030_PATCH'\t138394865\tStandard\n",
      "#\t\t'9'\t138394717\tStandard\n",
      "#\t\t'HSCHR11_2_CTG8'\t135117433\tStandard\n",
      "#\t\t'HG151_NOVEL_TEST'\t135097696\tStandard\n",
      "#\t\t'HSCHR11_1_CTG5'\t135097255\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_2'\t135096748\tStandard\n",
      "#\t\t'HG142_HG150_NOVEL_TEST'\t135094228\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1'\t135093444\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_1'\t135092355\tStandard\n",
      "#\t\t'HSCHR11_1_CTG6'\t135092332\tStandard\n",
      "#\t\t'HSCHR11_3_CTG1'\t135092180\tStandard\n",
      "#\t\t'HG2115_PATCH'\t135092089\tStandard\n",
      "#\t\t'HG2114_PATCH'\t135091163\tStandard\n",
      "#\t\t'HSCHR11_1_CTG3_1'\t135090845\tStandard\n",
      "#\t\t'HG2116_PATCH'\t135090698\tStandard\n",
      "#\t\t'HSCHR11_1_CTG3'\t135090275\tStandard\n",
      "#\t\t'HG2111_PATCH'\t135089248\tStandard\n",
      "#\t\t'HG28_PATCH'\t135088641\tStandard\n",
      "#\t\t'HG107_PATCH'\t135088590\tStandard\n",
      "#\t\t'HSCHR11_1_CTG8'\t135088433\tStandard\n",
      "#\t\t'HG1521_PATCH'\t135087279\tStandard\n",
      "#\t\t'11'\t135086622\tStandard\n",
      "#\t\t'HG2217_PATCH'\t135086621\tStandard\n",
      "#\t\t'HG1708_PATCH'\t135086620\tStandard\n",
      "#\t\t'HG2060_PATCH'\t135086593\tStandard\n",
      "#\t\t'HSCHR11_1_CTG7'\t135085484\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1_1'\t135083841\tStandard\n",
      "#\t\t'HSCHR11_1_CTG2'\t135083598\tStandard\n",
      "#\t\t'HG1445_PATCH'\t135063159\tStandard\n",
      "#\t\t'HSCHR10_1_CTG1'\t133844722\tStandard\n",
      "#\t\t'HG545_PATCH'\t133833661\tStandard\n",
      "#\t\t'HSCHR10_1_CTG4'\t133815819\tStandard\n",
      "#\t\t'HSCHR10_1_CTG2'\t133813485\tStandard\n",
      "#\t\t'HSCHR10_1_CTG6'\t133801239\tStandard\n",
      "#\t\t'HG2191_PATCH'\t133799479\tStandard\n",
      "#\t\t'HG2244_HG2245_PATCH'\t133797425\tStandard\n",
      "#\t\t'10'\t133797422\tStandard\n",
      "#\t\t'HG2334_PATCH'\t133797389\tStandard\n",
      "#\t\t'HG2242_HG2243_PATCH'\t133797297\tStandard\n",
      "#\t\t'HG2241_PATCH'\t133792722\tStandard\n",
      "#\t\t'HSCHR10_1_CTG3'\t133783218\tStandard\n",
      "#\t\t'HG1277_PATCH'\t133754853\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2'\t133307257\tStandard\n",
      "#\t\t'HSCHR12_9_CTG2_1'\t133289677\tStandard\n",
      "#\t\t'HSCHR12_7_CTG2_1'\t133289478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2_1'\t133284468\tStandard\n",
      "#\t\t'HSCHR12_8_CTG2_1'\t133282605\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2_1'\t133282358\tStandard\n",
      "#\t\t'HSCHR12_2_CTG1'\t133282237\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2_1'\t133281486\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2'\t133281368\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2_1'\t133281351\tStandard\n",
      "#\t\t'HG1815_PATCH'\t133281281\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2_1'\t133279478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG1'\t133279324\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2'\t133277582\tStandard\n",
      "#\t\t'HSCHR12_6_CTG2_1'\t133275361\tStandard\n",
      "#\t\t'HG1362_PATCH'\t133275314\tStandard\n",
      "#\t\t'12'\t133275309\tStandard\n",
      "#\t\t'HG2047_PATCH'\t133275304\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2'\t133275237\tStandard\n",
      "#\t\t'HG23_PATCH'\t133275184\tStandard\n",
      "#\t\t'HG2063_PATCH'\t133275140\tStandard\n",
      "#\t\t'HG2246_HG2248_HG2276_PATCH'\t133274640\tStandard\n",
      "#\t\t'HG1398_PATCH'\t133274540\tStandard\n",
      "#\t\t'HG2247_PATCH'\t133274000\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2'\t133271566\tStandard\n",
      "#\t\t'HG2291_PATCH'\t114534236\tStandard\n",
      "#\t\t'HG1523_PATCH'\t114458500\tStandard\n",
      "#\t\t'HG2216_PATCH'\t114376428\tStandard\n",
      "#\t\t'HSCHR13_1_CTG7'\t114374491\tStandard\n",
      "#\t\t'HSCHR13_1_CTG3'\t114370226\tStandard\n",
      "#\t\t'HG1817_1_PATCH'\t114369395\tStandard\n",
      "#\t\t'HSCHR13_1_CTG8'\t114368996\tStandard\n",
      "#\t\t'HSCHR13_1_CTG4'\t114368332\tStandard\n",
      "#\t\t'HSCHR13_1_CTG2'\t114368315\tStandard\n",
      "#\t\t'HSCHR13_1_CTG6'\t114367169\tStandard\n",
      "#\t\t'HSCHR13_1_CTG5'\t114365350\tStandard\n",
      "#\t\t'13'\t114364328\tStandard\n",
      "#\t\t'HG2509_PATCH'\t114364327\tStandard\n",
      "#\t\t'HG2249_PATCH'\t114364211\tStandard\n",
      "#\t\t'HG2288_HG2289_PATCH'\t114360394\tStandard\n",
      "#\t\t'HSCHR13_1_CTG1'\t114348172\tStandard\n",
      "#\t\t'HG1524_PATCH'\t114315104\tStandard\n",
      "#\t\t'HSCHR14_3_CTG1'\t107067894\tStandard\n",
      "#\t\t'HSCHR14_2_CTG1'\t107050862\tStandard\n",
      "#\t\t'HSCHR14_8_CTG1'\t107050407\tStandard\n",
      "#\t\t'HSCHR14_9_CTG1'\t107050362\tStandard\n",
      "#\t\t'HSCHR14_1_CTG1'\t107047377\tStandard\n",
      "#\t\t'HSCHR14_7_CTG1'\t107045728\tStandard\n",
      "#\t\t'14'\t107043718\tStandard\n",
      "#\t\t'HG1_PATCH'\t107043717\tStandard\n",
      "#\t\t'HG2510_PATCH'\t107043717\tStandard\n",
      "#\t\t'HG2365_PATCH'\t102714182\tStandard\n",
      "#\t\t'HSCHR15_4_CTG8'\t102071387\tStandard\n",
      "#\t\t'HSCHR15_5_CTG8'\t102070278\tStandard\n",
      "#\t\t'HSCHR15_2_CTG8'\t102003326\tStandard\n",
      "#\t\t'HSCHR15_1_CTG8'\t102002809\tStandard\n",
      "#\t\t'HG2499_PATCH'\t102001626\tStandard\n",
      "#\t\t'HSCHR15_3_CTG8'\t101999716\tStandard\n",
      "#\t\t'HSCHR15_1_CTG1'\t101998275\tStandard\n",
      "#\t\t'HSCHR15_2_CTG3'\t101995966\tStandard\n",
      "#\t\t'HSCHR15_1_CTG3'\t101994657\tStandard\n",
      "#\t\t'HSCHR15_3_CTG3'\t101993763\tStandard\n",
      "#\t\t'HG2198_PATCH'\t101993033\tStandard\n",
      "#\t\t'15'\t101991189\tStandard\n",
      "#\t\t'HG2511_PATCH'\t101991188\tStandard\n",
      "#\t\t'HSCHR15_6_CTG8'\t101521929\tStandard\n",
      "#\t\t'HG926_PATCH'\t90680781\tStandard\n",
      "#\t\t'HSCHR16_1_CTG1'\t90426449\tStandard\n",
      "#\t\t'HSCHR16_3_CTG3_1'\t90346170\tStandard\n",
      "#\t\t'HSCHR16_5_CTG1'\t90344275\tStandard\n",
      "#\t\t'HSCHR16_2_CTG3_1'\t90343477\tStandard\n",
      "#\t\t'HSCHR16_4_CTG3_1'\t90343133\tStandard\n",
      "#\t\t'HSCHR16_CTG2'\t90341867\tStandard\n",
      "#\t\t'HG2471_PATCH'\t90341128\tStandard\n",
      "#\t\t'HSCHR16_4_CTG1'\t90340369\tStandard\n",
      "#\t\t'HSCHR16_3_CTG1'\t90340304\tStandard\n",
      "#\t\t'HG2263_PATCH'\t90338583\tStandard\n",
      "#\t\t'HSCHR16_5_CTG3_1'\t90338377\tStandard\n",
      "#\t\t'16'\t90338345\tStandard\n",
      "#\t\t'HSCHR16_1_CTG3_1'\t90335239\tStandard\n",
      "#\t\t'HSCHR17_1_CTG5'\t83552665\tStandard\n",
      "#\t\t'HSCHR17_10_CTG4'\t83359087\tStandard\n",
      "#\t\t'HSCHR17_1_CTG1'\t83290495\tStandard\n",
      "#\t\t'HSCHR17_12_CTG4'\t83271781\tStandard\n",
      "#\t\t'HSCHR17_3_CTG4'\t83269063\tStandard\n",
      "#\t\t'HSCHR17_3_CTG1'\t83267046\tStandard\n",
      "#\t\t'HSCHR17_11_CTG4'\t83264946\tStandard\n",
      "#\t\t'HSCHR17_1_CTG4'\t83264766\tStandard\n",
      "#\t\t'HSCHR17_9_CTG4'\t83263791\tStandard\n",
      "#\t\t'HSCHR17_2_CTG4'\t83263494\tStandard\n",
      "#\t\t'HG2087_PATCH'\t83260478\tStandard\n",
      "#\t\t'HSCHR17_2_CTG1'\t83259496\tStandard\n",
      "#\t\t'HSCHR17_8_CTG4'\t83258474\tStandard\n",
      "#\t\t'HSCHR17_3_CTG2'\t83258211\tStandard\n",
      "#\t\t'HSCHR17_4_CTG4'\t83257896\tStandard\n",
      "#\t\t'17'\t83257441\tStandard\n",
      "#\t\t'HG2046_PATCH'\t83257439\tStandard\n",
      "#\t\t'HSCHR17_1_CTG2'\t83256362\tStandard\n",
      "#\t\t'HSCHR17_1_CTG9'\t83255504\tStandard\n",
      "#\t\t'HSCHR17_5_CTG4'\t83247119\tStandard\n",
      "#\t\t'HSCHR17_6_CTG4'\t83246071\tStandard\n",
      "#\t\t'HG2285_HG106_HG2252_PATCH'\t83219555\tStandard\n",
      "#\t\t'HG1320_PATCH'\t83207673\tStandard\n",
      "#\t\t'HSCHR17_2_CTG2'\t83160220\tStandard\n",
      "#\t\t'HSCHR17_2_CTG5'\t83086355\tStandard\n",
      "#\t\t'HSCHR17_7_CTG4'\t83058228\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2'\t80442734\tStandard\n",
      "#\t\t'HSCHR18_1_CTG1'\t80395815\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2'\t80386177\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2_1'\t80382671\tStandard\n",
      "#\t\t'HSCHR18_1_CTG1_1'\t80380517\tStandard\n",
      "#\t\t'HSCHR18_2_CTG1_1'\t80379367\tStandard\n",
      "#\t\t'HG2442_PATCH'\t80379182\tStandard\n",
      "#\t\t'HSCHR18_3_CTG2_1'\t80379039\tStandard\n",
      "#\t\t'HSCHR18_5_CTG1_1'\t80378965\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2_1'\t80378720\tStandard\n",
      "#\t\t'HSCHR18_4_CTG1_1'\t80378406\tStandard\n",
      "#\t\t'HG2213_PATCH'\t80377536\tStandard\n",
      "#\t\t'HSCHR18_ALT2_CTG2_1'\t80376883\tStandard\n",
      "#\t\t'HSCHR18_ALT21_CTG2_1'\t80376343\tStandard\n",
      "#\t\t'18'\t80373285\tStandard\n",
      "#\t\t'HG2412_PATCH'\t80373284\tStandard\n",
      "#\t\t'HSCHR20_1_CTG1'\t64450183\tStandard\n",
      "#\t\t'HSCHR20_1_CTG3'\t64447812\tStandard\n",
      "#\t\t'HSCHR20_1_CTG4'\t64446453\tStandard\n",
      "#\t\t'20'\t64444167\tStandard\n",
      "#\t\t'HSCHR20_1_CTG2'\t64441059\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TB04_CTG3_1'\t58750420\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_B_HAP_CTG3_1'\t58689284\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_B_HAP_CTG3_1'\t58689016\tStandard\n",
      "#\t\t'HSCHR19KIR_0019-4656-B_CTG3_1'\t58675214\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_BA2_HAP_CTG3_1'\t58662330\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_BA2_HAP_CTG3_1'\t58662237\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_BA2_HAP_CTG3_1'\t58661364\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_J_CTG3_1'\t58650772\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_BAX_HAP_CTG3_1'\t58647709\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TB01_CTG3_1'\t58645065\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_AB_HAP_CTG3_1'\t58644315\tStandard\n",
      "#\t\t'HG2021_PATCH'\t58632883\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_BDEL_HAP_CTG3_1'\t58631446\tStandard\n",
      "#\t\t'HSCHR19KIR_RP5_B_HAP_CTG3_1'\t58631402\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_BDEL_HAP_CTG3_1'\t58631206\tStandard\n",
      "#\t\t'HSCHR19_2_CTG2'\t58627095\tStandard\n",
      "#\t\t'HSCHR19_3_CTG3_1'\t58625958\tStandard\n",
      "#\t\t'HSCHR19_3_CTG2'\t58625823\tStandard\n",
      "#\t\t'HSCHR19_4_CTG3_1'\t58625731\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_S_CTG3_1'\t58625321\tStandard\n",
      "#\t\t'HSCHR19_1_CTG3_1'\t58624789\tStandard\n",
      "#\t\t'HSCHR19_1_CTG2'\t58624032\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_I_CTG3_1'\t58623235\tStandard\n",
      "#\t\t'HSCHR19_4_CTG2'\t58619803\tStandard\n",
      "#\t\t'HSCHR19_5_CTG2'\t58619095\tStandard\n",
      "#\t\t'HSCHR19_2_CTG3_1'\t58618601\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_C_P_CTG3_1'\t58618008\tStandard\n",
      "#\t\t'HG109_PATCH'\t58617934\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_A_HAP_CTG3_1'\t58617658\tStandard\n",
      "#\t\t'HG26_PATCH'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_A_HAP_CTG3_1'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_7191059-1_CTG3_1'\t58617649\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TA01_2_CTG3_1'\t58617647\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_T_P_CTG3_1'\t58617636\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_A_HAP_CTG3_1'\t58617634\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TA01_1_CTG3_1'\t58617632\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2394_CTG3_1'\t58617630\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_A1_HAP_CTG3_1'\t58617625\tStandard\n",
      "#\t\t'19'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_A_HAP_CTG3_1'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_502960008-1_CTG3_1'\t58617609\tStandard\n",
      "#\t\t'HSCHR19KIR_7191059-2_CTG3_1'\t58617604\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_A_HAP_CTG3_1'\t58617587\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_A_HAP_CTG3_1'\t58617473\tStandard\n",
      "#\t\t'HSCHR19KIR_0019-4656-A_CTG3_1'\t58617441\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_A_HAP_CTG3_1'\t58616795\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_A_HAP_CTG3_1'\t58616725\tStandard\n",
      "#\t\t'HSCHR19KIR_502960008-2_CTG3_1'\t58616590\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_BA1_HAP_CTG3_1'\t58603994\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2396_CTG3_1'\t58603982\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_BA1_HAP_CTG3_1'\t58603800\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_BA1_HAP_CTG3_1'\t58602595\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2393_CTG3_1'\t58583172\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_T_CTG3_1'\t58561614\tStandard\n",
      "#\t\t'HSCHR19LRC_COX1_CTG3_1'\t58546647\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF1_CTG3_1'\t58546031\tStandard\n",
      "#\t\t'HSCHR19KIR_0010-5217-AB_CTG3_1'\t58538377\tStandard\n",
      "#\t\t'HSCHR19KIR_CA04_CTG3_1'\t58537114\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF2_CTG3_1'\t58355410\tStandard\n",
      "#\t\t'HSCHR19LRC_COX2_CTG3_1'\t58288451\tStandard\n",
      "#\t\t'HG2062_PATCH'\t57262172\tStandard\n",
      "#\t\t'Y'\t57227415\tStandard\n",
      "#\t\t'HG1535_PATCH'\t57187419\tStandard\n",
      "#\t\t'HG1531_PATCH'\t57177176\tStandard\n",
      "#\t\t'HG1485_PATCH'\t50915261\tStandard\n",
      "#\t\t'HSCHR22_1_CTG7'\t50872732\tStandard\n",
      "#\t\t'HSCHR22_7_CTG1'\t50845852\tStandard\n",
      "#\t\t'HSCHR22_1_CTG6'\t50842710\tStandard\n",
      "#\t\t'HSCHR22_5_CTG1'\t50832153\tStandard\n",
      "#\t\t'HSCHR22_8_CTG1'\t50832130\tStandard\n",
      "#\t\t'HSCHR22_2_CTG1'\t50830603\tStandard\n",
      "#\t\t'HSCHR22_1_CTG3'\t50827573\tStandard\n",
      "#\t\t'HSCHR22_1_CTG4'\t50824699\tStandard\n",
      "#\t\t'HSCHR22_1_CTG5'\t50821308\tStandard\n",
      "#\t\t'HG1311_PATCH'\t50818486\tStandard\n",
      "#\t\t'HG494_PATCH'\t50818473\tStandard\n",
      "#\t\t'22'\t50818468\tStandard\n",
      "#\t\t'HSCHR22_3_CTG1'\t50818468\tStandard\n",
      "#\t\t'HG2512_PATCH'\t50818467\tStandard\n",
      "#\t\t'HSCHR22_6_CTG1'\t50815919\tStandard\n",
      "#\t\t'HSCHR22_1_CTG1'\t50806372\tStandard\n",
      "#\t\t'HSCHR22_4_CTG1'\t50806321\tStandard\n",
      "#\t\t'HSCHR22_1_CTG2'\t50788525\tStandard\n",
      "#\t\t'HG2513_PATCH'\t47068079\tStandard\n",
      "#\t\t'HSCHR21_1_CTG1_1'\t46722569\tStandard\n",
      "#\t\t'HSCHR21_4_CTG1_1'\t46719541\tStandard\n",
      "#\t\t'HSCHR21_2_CTG1_1'\t46716685\tStandard\n",
      "#\t\t'HSCHR21_3_CTG1_1'\t46715777\tStandard\n",
      "#\t\t'HSCHR21_6_CTG1_1'\t46713516\tStandard\n",
      "#\t\t'HSCHR21_8_CTG1_1'\t46713185\tStandard\n",
      "#\t\t'HSCHR21_5_CTG2'\t46712294\tStandard\n",
      "#\t\t'21'\t46709983\tStandard\n",
      "#\t\t'KI270728.1'\t1872759\tStandard\n",
      "#\t\t'KI270727.1'\t448248\tStandard\n",
      "#\t\t'KI270442.1'\t392061\tStandard\n",
      "#\t\t'KI270729.1'\t280839\tStandard\n",
      "#\t\t'GL000225.1'\t211173\tStandard\n",
      "#\t\t'KI270743.1'\t210658\tStandard\n",
      "#\t\t'GL000008.2'\t209709\tStandard\n",
      "#\t\t'GL000009.2'\t201709\tStandard\n",
      "#\t\t'KI270747.1'\t198735\tStandard\n",
      "#\t\t'KI270722.1'\t194050\tStandard\n",
      "#\t\t'GL000194.1'\t191469\tStandard\n",
      "#\t\t'KI270742.1'\t186739\tStandard\n",
      "#\t\t'GL000205.2'\t185591\tStandard\n",
      "#\t\t'GL000195.1'\t182896\tStandard\n",
      "#\t\t'KI270736.1'\t181920\tStandard\n",
      "#\t\t'KI270733.1'\t179772\tStandard\n",
      "#\t\t'GL000224.1'\t179693\tStandard\n",
      "#\t\t'GL000219.1'\t179198\tStandard\n",
      "#\t\t'KI270719.1'\t176845\tStandard\n",
      "#\t\t'GL000216.2'\t176608\tStandard\n",
      "#\t\t'KI270712.1'\t176043\tStandard\n",
      "#\t\t'KI270706.1'\t175055\tStandard\n",
      "#\t\t'KI270725.1'\t172810\tStandard\n",
      "#\t\t'KI270744.1'\t168472\tStandard\n",
      "#\t\t'KI270734.1'\t165050\tStandard\n",
      "#\t\t'GL000213.1'\t164239\tStandard\n",
      "#\t\t'GL000220.1'\t161802\tStandard\n",
      "#\t\t'KI270715.1'\t161471\tStandard\n",
      "#\t\t'GL000218.1'\t161147\tStandard\n",
      "#\t\t'KI270749.1'\t158759\tStandard\n",
      "#\t\t'KI270741.1'\t157432\tStandard\n",
      "#\t\t'GL000221.1'\t155397\tStandard\n",
      "#\t\t'KI270716.1'\t153799\tStandard\n",
      "#\t\t'KI270731.1'\t150754\tStandard\n",
      "#\t\t'KI270751.1'\t150742\tStandard\n",
      "#\t\t'KI270750.1'\t148850\tStandard\n",
      "#\t\t'KI270519.1'\t138126\tStandard\n",
      "#\t\t'GL000214.1'\t137718\tStandard\n",
      "#\t\t'KI270708.1'\t127682\tStandard\n",
      "#\t\t'KI270730.1'\t112551\tStandard\n",
      "#\t\t'KI270438.1'\t112505\tStandard\n",
      "#\t\t'KI270737.1'\t103838\tStandard\n",
      "#\t\t'KI270721.1'\t100316\tStandard\n",
      "#\t\t'KI270738.1'\t99375\tStandard\n",
      "#\t\t'KI270748.1'\t93321\tStandard\n",
      "#\t\t'KI270435.1'\t92983\tStandard\n",
      "#\t\t'GL000208.1'\t92689\tStandard\n",
      "#\t\t'KI270538.1'\t91309\tStandard\n",
      "#\t\t'KI270756.1'\t79590\tStandard\n",
      "#\t\t'KI270739.1'\t73985\tStandard\n",
      "#\t\t'KI270757.1'\t71251\tStandard\n",
      "#\t\t'KI270709.1'\t66860\tStandard\n",
      "#\t\t'KI270746.1'\t66486\tStandard\n",
      "#\t\t'KI270753.1'\t62944\tStandard\n",
      "#\t\t'KI270589.1'\t44474\tStandard\n",
      "#\t\t'KI270726.1'\t43739\tStandard\n",
      "#\t\t'KI270735.1'\t42811\tStandard\n",
      "#\t\t'KI270711.1'\t42210\tStandard\n",
      "#\t\t'KI270745.1'\t41891\tStandard\n",
      "#\t\t'KI270714.1'\t41717\tStandard\n",
      "#\t\t'KI270732.1'\t41543\tStandard\n",
      "#\t\t'KI270713.1'\t40745\tStandard\n",
      "#\t\t'KI270754.1'\t40191\tStandard\n",
      "#\t\t'KI270710.1'\t40176\tStandard\n",
      "#\t\t'KI270717.1'\t40062\tStandard\n",
      "#\t\t'KI270724.1'\t39555\tStandard\n",
      "#\t\t'KI270720.1'\t39050\tStandard\n",
      "#\t\t'KI270723.1'\t38115\tStandard\n",
      "#\t\t'KI270718.1'\t38054\tStandard\n",
      "#\t\t'KI270317.1'\t37690\tStandard\n",
      "#\t\t'KI270740.1'\t37240\tStandard\n",
      "#\t\t'KI270755.1'\t36723\tStandard\n",
      "#\t\t'KI270707.1'\t32032\tStandard\n",
      "#\t\t'KI270579.1'\t31033\tStandard\n",
      "#\t\t'KI270752.1'\t27745\tStandard\n",
      "#\t\t'KI270512.1'\t22689\tStandard\n",
      "#\t\t'KI270322.1'\t21476\tStandard\n",
      "#\t\t'MT'\t16569\tStandard\n",
      "#\t\t'GL000226.1'\t15008\tStandard\n",
      "#\t\t'KI270311.1'\t12399\tStandard\n",
      "#\t\t'KI270366.1'\t8320\tStandard\n",
      "#\t\t'KI270511.1'\t8127\tStandard\n",
      "#\t\t'KI270448.1'\t7992\tStandard\n",
      "#\t\t'KI270521.1'\t7642\tStandard\n",
      "#\t\t'KI270581.1'\t7046\tStandard\n",
      "#\t\t'KI270582.1'\t6504\tStandard\n",
      "#\t\t'KI270515.1'\t6361\tStandard\n",
      "#\t\t'KI270588.1'\t6158\tStandard\n",
      "#\t\t'KI270591.1'\t5796\tStandard\n",
      "#\t\t'KI270522.1'\t5674\tStandard\n",
      "#\t\t'KI270507.1'\t5353\tStandard\n",
      "#\t\t'KI270590.1'\t4685\tStandard\n",
      "#\t\t'KI270584.1'\t4513\tStandard\n",
      "#\t\t'KI270320.1'\t4416\tStandard\n",
      "#\t\t'KI270382.1'\t4215\tStandard\n",
      "#\t\t'KI270468.1'\t4055\tStandard\n",
      "#\t\t'KI270467.1'\t3920\tStandard\n",
      "#\t\t'KI270362.1'\t3530\tStandard\n",
      "#\t\t'KI270517.1'\t3253\tStandard\n",
      "#\t\t'KI270593.1'\t3041\tStandard\n",
      "#\t\t'KI270528.1'\t2983\tStandard\n",
      "#\t\t'KI270587.1'\t2969\tStandard\n",
      "#\t\t'KI270364.1'\t2855\tStandard\n",
      "#\t\t'KI270371.1'\t2805\tStandard\n",
      "#\t\t'KI270333.1'\t2699\tStandard\n",
      "#\t\t'KI270374.1'\t2656\tStandard\n",
      "#\t\t'KI270411.1'\t2646\tStandard\n",
      "#\t\t'KI270414.1'\t2489\tStandard\n",
      "#\t\t'KI270510.1'\t2415\tStandard\n",
      "#\t\t'KI270390.1'\t2387\tStandard\n",
      "#\t\t'KI270375.1'\t2378\tStandard\n",
      "#\t\t'KI270420.1'\t2321\tStandard\n",
      "#\t\t'KI270509.1'\t2318\tStandard\n",
      "#\t\t'KI270315.1'\t2276\tStandard\n",
      "#\t\t'KI270302.1'\t2274\tStandard\n",
      "#\t\t'KI270518.1'\t2186\tStandard\n",
      "#\t\t'KI270530.1'\t2168\tStandard\n",
      "#\t\t'KI270304.1'\t2165\tStandard\n",
      "#\t\t'KI270418.1'\t2145\tStandard\n",
      "#\t\t'KI270424.1'\t2140\tStandard\n",
      "#\t\t'KI270417.1'\t2043\tStandard\n",
      "#\t\t'KI270508.1'\t1951\tStandard\n",
      "#\t\t'KI270303.1'\t1942\tStandard\n",
      "#\t\t'KI270381.1'\t1930\tStandard\n",
      "#\t\t'KI270529.1'\t1899\tStandard\n",
      "#\t\t'KI270425.1'\t1884\tStandard\n",
      "#\t\t'KI270396.1'\t1880\tStandard\n",
      "#\t\t'KI270363.1'\t1803\tStandard\n",
      "#\t\t'KI270386.1'\t1788\tStandard\n",
      "#\t\t'KI270465.1'\t1774\tStandard\n",
      "#\t\t'KI270383.1'\t1750\tStandard\n",
      "#\t\t'KI270384.1'\t1658\tStandard\n",
      "#\t\t'KI270330.1'\t1652\tStandard\n",
      "#\t\t'KI270372.1'\t1650\tStandard\n",
      "#\t\t'KI270548.1'\t1599\tStandard\n",
      "#\t\t'KI270580.1'\t1553\tStandard\n",
      "#\t\t'KI270387.1'\t1537\tStandard\n",
      "#\t\t'KI270391.1'\t1484\tStandard\n",
      "#\t\t'KI270305.1'\t1472\tStandard\n",
      "#\t\t'KI270373.1'\t1451\tStandard\n",
      "#\t\t'KI270422.1'\t1445\tStandard\n",
      "#\t\t'KI270316.1'\t1444\tStandard\n",
      "#\t\t'KI270338.1'\t1428\tStandard\n",
      "#\t\t'KI270340.1'\t1428\tStandard\n",
      "#\t\t'KI270583.1'\t1400\tStandard\n",
      "#\t\t'KI270334.1'\t1368\tStandard\n",
      "#\t\t'KI270429.1'\t1361\tStandard\n",
      "#\t\t'KI270393.1'\t1308\tStandard\n",
      "#\t\t'KI270516.1'\t1300\tStandard\n",
      "#\t\t'KI270389.1'\t1298\tStandard\n",
      "#\t\t'KI270466.1'\t1233\tStandard\n",
      "#\t\t'KI270388.1'\t1216\tStandard\n",
      "#\t\t'KI270544.1'\t1202\tStandard\n",
      "#\t\t'KI270310.1'\t1201\tStandard\n",
      "#\t\t'KI270412.1'\t1179\tStandard\n",
      "#\t\t'KI270395.1'\t1143\tStandard\n",
      "#\t\t'KI270376.1'\t1136\tStandard\n",
      "#\t\t'KI270337.1'\t1121\tStandard\n",
      "#\t\t'KI270335.1'\t1048\tStandard\n",
      "#\t\t'KI270378.1'\t1048\tStandard\n",
      "#\t\t'KI270379.1'\t1045\tStandard\n",
      "#\t\t'KI270329.1'\t1040\tStandard\n",
      "#\t\t'KI270419.1'\t1029\tStandard\n",
      "#\t\t'KI270336.1'\t1026\tStandard\n",
      "#\t\t'KI270312.1'\t998\tStandard\n",
      "#\t\t'KI270539.1'\t993\tStandard\n",
      "#\t\t'KI270385.1'\t990\tStandard\n",
      "#\t\t'KI270423.1'\t981\tStandard\n",
      "#\t\t'KI270392.1'\t971\tStandard\n",
      "#\t\t'KI270394.1'\t970\tStandard\n",
      "#-----------------------------------------------\n",
      "\n",
      "00:00:21 Predicting variants\n",
      "00:00:21 Loading sequences for chromosome '12' from file '/home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/./data/GRCh38.99/sequence.12.bin'\n",
      "00:00:22 Building sequence tree for chromosome '12'\n",
      "00:00:22 Done. Loaded 1488 sequences.\n",
      "\n",
      "WARNINGS: Some warning were detected\n",
      "Warning type\tNumber of warnings\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON\t4\n",
      "\n",
      "\n",
      "00:00:22 Creating summary file: snpEff_summary.html\n",
      "00:00:23 Creating genes file: snpEff_genes.txt\n",
      "00:00:23 done.\n",
      "00:00:23 Done.\n",
      "00:00:00 SnpEff version SnpEff 5.3a (build 2025-09-02 10:24), by Pablo Cingolani\n",
      "00:00:00 Command: 'ann'\n",
      "00:00:00 Reading configuration file 'snpEff.config'. Genome: 'GRCh38.99'\n",
      "00:00:00 Looking for config file: '/home/mgpersonal/ctdna_kras_project/results/vcf/snpEff.config'\n",
      "00:00:00 Reading config file: /home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/snpEff.config\n",
      "00:00:00 done\n",
      "00:00:00 Reading database for genome version 'GRCh38.99' from file '/home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/./data/GRCh38.99/snpEffectPredictor.bin' (this might take a while)\n",
      "00:00:15 done\n",
      "00:00:15 Loading Motifs and PWMs\n",
      "00:00:15 Building interval forest\n",
      "00:00:20 done.\n",
      "00:00:20 Genome stats :\n",
      "#-----------------------------------------------\n",
      "# Genome name                : 'Homo_sapiens'\n",
      "# Genome version             : 'GRCh38.99'\n",
      "# Genome ID                  : 'GRCh38.99[0]'\n",
      "# Has protein coding info    : true\n",
      "# Has Tr. Support Level info : true\n",
      "# Genes                      : 60676\n",
      "# Protein coding genes       : 20135\n",
      "#-----------------------------------------------\n",
      "# Transcripts                : 227818\n",
      "# Avg. transcripts per gene  : 3.75\n",
      "# TSL transcripts            : 169199\n",
      "#-----------------------------------------------\n",
      "# Checked transcripts        : \n",
      "#               AA sequences : 100576 ( 119.03% )\n",
      "#              DNA sequences : 172288 ( 75.63% )\n",
      "#-----------------------------------------------\n",
      "# Protein coding transcripts : 84494\n",
      "#              Length errors :  13469 ( 15.94% )\n",
      "#  STOP codons in CDS errors :     34 ( 0.04% )\n",
      "#         START codon errors :   8149 ( 9.64% )\n",
      "#        STOP codon warnings :   7264 ( 8.60% )\n",
      "#              UTR sequences :  81637 ( 35.83% )\n",
      "#               Total Errors :  20119 ( 23.81% )\n",
      "#-----------------------------------------------\n",
      "# Cds                        : 764354\n",
      "# Exons                      : 1376499\n",
      "# Exons with sequence        : 1376499\n",
      "# Exons without sequence     : 0\n",
      "# Avg. exons per transcript  : 6.04\n",
      "# WARNING!                   : Mitochondrion chromosome 'MT' does not have a mitochondrion codon table (codon table = 'Standard'). You should update the config file.\n",
      "#-----------------------------------------------\n",
      "# Number of chromosomes      : 639\n",
      "# Chromosomes                : Format 'chromo_name size codon_table'\n",
      "#\t\t'HSCHR1_2_CTG3'\t248975002\tStandard\n",
      "#\t\t'HSCHR1_1_CTG31'\t248973653\tStandard\n",
      "#\t\t'HSCHR1_2_CTG31'\t248971826\tStandard\n",
      "#\t\t'HSCHR1_1_CTG3'\t248968406\tStandard\n",
      "#\t\t'HSCHR1_3_CTG31'\t248967909\tStandard\n",
      "#\t\t'HSCHR1_9_CTG3'\t248965380\tStandard\n",
      "#\t\t'HSCHR1_4_CTG3'\t248963886\tStandard\n",
      "#\t\t'HSCHR1_2_CTG32_1'\t248962488\tStandard\n",
      "#\t\t'HSCHR1_8_CTG3'\t248962151\tStandard\n",
      "#\t\t'HSCHR1_3_CTG3'\t248961700\tStandard\n",
      "#\t\t'HSCHR1_1_CTG11'\t248961539\tStandard\n",
      "#\t\t'HSCHR1_3_CTG32_1'\t248960761\tStandard\n",
      "#\t\t'HSCHR1_5_CTG32_1'\t248960213\tStandard\n",
      "#\t\t'HSCHR1_1_CTG32_1'\t248960143\tStandard\n",
      "#\t\t'HG1832_PATCH'\t248959779\tStandard\n",
      "#\t\t'HG2095_PATCH'\t248959131\tStandard\n",
      "#\t\t'HSCHR1_ALT2_1_CTG32_1'\t248958866\tStandard\n",
      "#\t\t'HSCHR1_4_CTG32_1'\t248958766\tStandard\n",
      "#\t\t'HG2104_PATCH'\t248958259\tStandard\n",
      "#\t\t'HG986_PATCH'\t248956639\tStandard\n",
      "#\t\t'1'\t248956422\tStandard\n",
      "#\t\t'HG2058_PATCH'\t248956422\tStandard\n",
      "#\t\t'HSCHR1_6_CTG3'\t248955842\tStandard\n",
      "#\t\t'HG460_PATCH'\t248955838\tStandard\n",
      "#\t\t'HSCHR1_4_CTG31'\t248955631\tStandard\n",
      "#\t\t'HG2002_PATCH'\t248942227\tStandard\n",
      "#\t\t'HG1342_HG2282_PATCH'\t248929249\tStandard\n",
      "#\t\t'HSCHR1_5_CTG3'\t248926662\tStandard\n",
      "#\t\t'HG2290_PATCH'\t242235269\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7'\t242223512\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7_2'\t242210518\tStandard\n",
      "#\t\t'HSCHR2_7_CTG7_2'\t242202531\tStandard\n",
      "#\t\t'HSCHR2_5_CTG7_2'\t242199685\tStandard\n",
      "#\t\t'HSCHR2_8_CTG7_2'\t242199670\tStandard\n",
      "#\t\t'HSCHR2_2_CTG7_2'\t242199609\tStandard\n",
      "#\t\t'HSCHR2_1_CTG5'\t242199562\tStandard\n",
      "#\t\t'HSCHR2_1_CTG15'\t242198904\tStandard\n",
      "#\t\t'HSCHR2_2_CTG15'\t242198544\tStandard\n",
      "#\t\t'HSCHR2_2_CTG1'\t242198346\tStandard\n",
      "#\t\t'HSCHR2_6_CTG7_2'\t242198149\tStandard\n",
      "#\t\t'HSCHR2_4_CTG7_2'\t242196589\tStandard\n",
      "#\t\t'HSCHR2_3_CTG1'\t242195626\tStandard\n",
      "#\t\t'HG1384_PATCH'\t242195459\tStandard\n",
      "#\t\t'HSCHR2_1_CTG1'\t242194896\tStandard\n",
      "#\t\t'HSCHR2_3_CTG7_2'\t242194557\tStandard\n",
      "#\t\t'HSCHR2_3_CTG15'\t242194292\tStandard\n",
      "#\t\t'HG721_PATCH'\t242194216\tStandard\n",
      "#\t\t'HSCHR2_1_CTG7'\t242193540\tStandard\n",
      "#\t\t'HG2233_PATCH'\t242193535\tStandard\n",
      "#\t\t'2'\t242193529\tStandard\n",
      "#\t\t'HG2232_PATCH'\t242193529\tStandard\n",
      "#\t\t'HSCHR2_4_CTG1'\t242191076\tStandard\n",
      "#\t\t'HSCHR3_4_CTG1'\t198310749\tStandard\n",
      "#\t\t'HSCHR3_1_CTG2_1'\t198306097\tStandard\n",
      "#\t\t'HSCHR3_5_CTG1'\t198305844\tStandard\n",
      "#\t\t'HSCHR3_6_CTG2_1'\t198305007\tStandard\n",
      "#\t\t'HSCHR3_1_CTG3'\t198302882\tStandard\n",
      "#\t\t'HSCHR3_9_CTG2_1'\t198302489\tStandard\n",
      "#\t\t'HSCHR3_8_CTG2_1'\t198302280\tStandard\n",
      "#\t\t'HSCHR3_4_CTG2_1'\t198301383\tStandard\n",
      "#\t\t'HSCHR3_7_CTG2_1'\t198301024\tStandard\n",
      "#\t\t'HSCHR3_5_CTG2_1'\t198300594\tStandard\n",
      "#\t\t'HSCHR3_3_CTG2_1'\t198300421\tStandard\n",
      "#\t\t'HSCHR3_3_CTG1'\t198300110\tStandard\n",
      "#\t\t'HG2133_PATCH'\t198299611\tStandard\n",
      "#\t\t'HSCHR3_2_CTG2_1'\t198298736\tStandard\n",
      "#\t\t'HSCHR3_1_CTG1'\t198297480\tStandard\n",
      "#\t\t'HSCHR3_2_CTG3'\t198296961\tStandard\n",
      "#\t\t'HSCHR3_4_CTG3'\t198296867\tStandard\n",
      "#\t\t'HG2022_PATCH'\t198295941\tStandard\n",
      "#\t\t'HSCHR3_8_CTG3'\t198295934\tStandard\n",
      "#\t\t'HG126_PATCH'\t198295908\tStandard\n",
      "#\t\t'HG2235_PATCH'\t198295655\tStandard\n",
      "#\t\t'3'\t198295559\tStandard\n",
      "#\t\t'HG2066_PATCH'\t198295555\tStandard\n",
      "#\t\t'HG2236_PATCH'\t198295537\tStandard\n",
      "#\t\t'HSCHR3_6_CTG3'\t198294921\tStandard\n",
      "#\t\t'HSCHR3_7_CTG3'\t198294497\tStandard\n",
      "#\t\t'HSCHR3_5_CTG3'\t198293785\tStandard\n",
      "#\t\t'HSCHR3_3_CTG3'\t198293223\tStandard\n",
      "#\t\t'HSCHR3_9_CTG3'\t198291952\tStandard\n",
      "#\t\t'HG2237_PATCH'\t198290329\tStandard\n",
      "#\t\t'HG2525_PATCH'\t190293418\tStandard\n",
      "#\t\t'HSCHR4_11_CTG12'\t190233769\tStandard\n",
      "#\t\t'HSCHR4_1_CTG12'\t190227204\tStandard\n",
      "#\t\t'HSCHR4_7_CTG12'\t190225141\tStandard\n",
      "#\t\t'HSCHR4_1_CTG6'\t190224838\tStandard\n",
      "#\t\t'HSCHR4_1_CTG4'\t190221757\tStandard\n",
      "#\t\t'HSCHR4_12_CTG12'\t190221648\tStandard\n",
      "#\t\t'HSCHR4_4_CTG12'\t190220704\tStandard\n",
      "#\t\t'HSCHR4_5_CTG12'\t190219562\tStandard\n",
      "#\t\t'HSCHR4_9_CTG12'\t190219005\tStandard\n",
      "#\t\t'HSCHR4_1_CTG8_1'\t190218838\tStandard\n",
      "#\t\t'HSCHR4_6_CTG12'\t190218533\tStandard\n",
      "#\t\t'HSCHR4_8_CTG12'\t190218480\tStandard\n",
      "#\t\t'HSCHR4_3_CTG12'\t190218028\tStandard\n",
      "#\t\t'HSCHR4_2_CTG4'\t190217898\tStandard\n",
      "#\t\t'4'\t190214555\tStandard\n",
      "#\t\t'HG699_PATCH'\t190211220\tStandard\n",
      "#\t\t'HG1296_PATCH'\t190208697\tStandard\n",
      "#\t\t'HSCHR4_2_CTG12'\t190206797\tStandard\n",
      "#\t\t'HG1299_PATCH'\t190201596\tStandard\n",
      "#\t\t'HG1298_PATCH'\t190196285\tStandard\n",
      "#\t\t'HG705_PATCH'\t190172551\tStandard\n",
      "#\t\t'HG2023_PATCH'\t190147504\tStandard\n",
      "#\t\t'HSCHR4_1_CTG9'\t190096932\tStandard\n",
      "#\t\t'HSCHR5_8_CTG1'\t181668817\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1'\t181558402\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1_1'\t181554865\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1'\t181547139\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1_1'\t181546281\tStandard\n",
      "#\t\t'HSCHR5_7_CTG1'\t181544732\tStandard\n",
      "#\t\t'HSCHR5_1_CTG5'\t181543736\tStandard\n",
      "#\t\t'HSCHR5_9_CTG1'\t181543440\tStandard\n",
      "#\t\t'HSCHR5_3_CTG1'\t181542495\tStandard\n",
      "#\t\t'HSCHR5_2_CTG5'\t181542206\tStandard\n",
      "#\t\t'HSCHR5_6_CTG1'\t181540245\tStandard\n",
      "#\t\t'HSCHR5_5_CTG1'\t181539933\tStandard\n",
      "#\t\t'HG30_PATCH'\t181539112\tStandard\n",
      "#\t\t'HSCHR5_3_CTG5'\t181538569\tStandard\n",
      "#\t\t'5'\t181538259\tStandard\n",
      "#\t\t'HG1395_PATCH'\t181537525\tStandard\n",
      "#\t\t'HSCHR5_4_CTG1'\t181537434\tStandard\n",
      "#\t\t'HSCHR5_1_CTG1_1'\t180863570\tStandard\n",
      "#\t\t'HSCHR5_2_CTG1_1'\t180753562\tStandard\n",
      "#\t\t'HSCHR6_MHC_MCF_CTG1'\t170976334\tStandard\n",
      "#\t\t'HSCHR6_MHC_MANN_CTG1'\t170959829\tStandard\n",
      "#\t\t'HSCHR6_MHC_SSTO_CTG1'\t170946136\tStandard\n",
      "#\t\t'HSCHR6_MHC_APD_CTG1'\t170845044\tStandard\n",
      "#\t\t'HSCHR6_1_CTG3'\t170836231\tStandard\n",
      "#\t\t'HSCHR6_1_CTG2'\t170833811\tStandard\n",
      "#\t\t'HSCHR6_1_CTG6'\t170818797\tStandard\n",
      "#\t\t'HG2072_PATCH'\t170816182\tStandard\n",
      "#\t\t'HSCHR6_1_CTG9'\t170815583\tStandard\n",
      "#\t\t'HSCHR6_1_CTG5'\t170815302\tStandard\n",
      "#\t\t'HSCHR6_8_CTG1'\t170813052\tStandard\n",
      "#\t\t'HSCHR6_1_CTG7'\t170812567\tStandard\n",
      "#\t\t'HG2128_PATCH'\t170811684\tStandard\n",
      "#\t\t'HSCHR6_1_CTG10'\t170810830\tStandard\n",
      "#\t\t'HSCHR6_1_CTG8'\t170810115\tStandard\n",
      "#\t\t'HSCHR6_1_CTG4'\t170808607\tStandard\n",
      "#\t\t'HG2121_PATCH'\t170808322\tStandard\n",
      "#\t\t'6'\t170805979\tStandard\n",
      "#\t\t'HG2057_PATCH'\t170805979\tStandard\n",
      "#\t\t'HSCHR6_MHC_DBB_CTG1'\t170783898\tStandard\n",
      "#\t\t'HG1651_PATCH'\t170779650\tStandard\n",
      "#\t\t'HG563_PATCH'\t170759097\tStandard\n",
      "#\t\t'HSCHR6_MHC_QBL_CTG1'\t170734801\tStandard\n",
      "#\t\t'HSCHR6_MHC_COX_CTG1'\t170727598\tStandard\n",
      "#\t\t'HSCHR7_3_CTG4_4'\t159441130\tStandard\n",
      "#\t\t'HSCHR7_2_CTG6'\t159407160\tStandard\n",
      "#\t\t'HG1309_PATCH'\t159365091\tStandard\n",
      "#\t\t'HSCHR7_3_CTG1'\t159352526\tStandard\n",
      "#\t\t'HSCHR7_1_CTG6'\t159351981\tStandard\n",
      "#\t\t'HG2088_PATCH'\t159351667\tStandard\n",
      "#\t\t'HSCHR7_2_CTG7'\t159349352\tStandard\n",
      "#\t\t'HSCHR7_1_CTG1'\t159349278\tStandard\n",
      "#\t\t'HSCHR7_1_CTG4_4'\t159348604\tStandard\n",
      "#\t\t'HSCHR7_2_CTG1'\t159348294\tStandard\n",
      "#\t\t'HSCHR7_1_CTG7'\t159346281\tStandard\n",
      "#\t\t'HG2266_PATCH'\t159346106\tStandard\n",
      "#\t\t'7'\t159345973\tStandard\n",
      "#\t\t'HG2239_PATCH'\t159345319\tStandard\n",
      "#\t\t'HSCHR7_3_CTG6'\t159344441\tStandard\n",
      "#\t\t'HSCHR7_2_CTG4_4'\t159333789\tStandard\n",
      "#\t\t'HG708_PATCH'\t159270839\tStandard\n",
      "#\t\t'HSCHRX_2_CTG12'\t156053172\tStandard\n",
      "#\t\t'HSCHRX_3_CTG7'\t156050387\tStandard\n",
      "#\t\t'HSCHRX_1_CTG3'\t156043585\tStandard\n",
      "#\t\t'X'\t156040895\tStandard\n",
      "#\t\t'HSCHRX_2_CTG3'\t156039875\tStandard\n",
      "#\t\t'HG1507_PATCH'\t155996656\tStandard\n",
      "#\t\t'HG1506_PATCH'\t155994589\tStandard\n",
      "#\t\t'HG1509_PATCH'\t155994530\tStandard\n",
      "#\t\t'HG439_PATCH'\t155960230\tStandard\n",
      "#\t\t'HG1466_PATCH'\t155848651\tStandard\n",
      "#\t\t'HSCHR8_1_CTG6'\t145150400\tStandard\n",
      "#\t\t'HSCHR8_7_CTG7'\t145147686\tStandard\n",
      "#\t\t'HSCHR8_5_CTG7'\t145144538\tStandard\n",
      "#\t\t'HG2067_PATCH'\t145144434\tStandard\n",
      "#\t\t'HSCHR8_4_CTG7'\t145143933\tStandard\n",
      "#\t\t'HSCHR8_1_CTG7'\t145142033\tStandard\n",
      "#\t\t'HSCHR8_2_CTG7'\t145141868\tStandard\n",
      "#\t\t'HSCHR8_6_CTG7'\t145141837\tStandard\n",
      "#\t\t'HSCHR8_2_CTG1'\t145141567\tStandard\n",
      "#\t\t'HSCHR8_4_CTG1'\t145141010\tStandard\n",
      "#\t\t'HG2068_PATCH'\t145140411\tStandard\n",
      "#\t\t'8'\t145138636\tStandard\n",
      "#\t\t'HG2419_PATCH'\t145138635\tStandard\n",
      "#\t\t'HSCHR8_7_CTG1'\t145138198\tStandard\n",
      "#\t\t'HSCHR8_5_CTG1'\t145137619\tStandard\n",
      "#\t\t'HSCHR8_6_CTG1'\t145136257\tStandard\n",
      "#\t\t'HSCHR8_3_CTG1'\t145130612\tStandard\n",
      "#\t\t'HSCHR8_3_CTG7'\t145125643\tStandard\n",
      "#\t\t'HSCHR8_1_CTG1'\t145124952\tStandard\n",
      "#\t\t'HSCHR8_8_CTG1'\t145123326\tStandard\n",
      "#\t\t'HSCHR8_9_CTG1'\t144983589\tStandard\n",
      "#\t\t'HG76_PATCH'\t144938989\tStandard\n",
      "#\t\t'HSCHR9_1_CTG1'\t138409529\tStandard\n",
      "#\t\t'HSCHR9_1_CTG4'\t138405815\tStandard\n",
      "#\t\t'HSCHR9_1_CTG5'\t138404305\tStandard\n",
      "#\t\t'HSCHR9_1_CTG2'\t138402913\tStandard\n",
      "#\t\t'HSCHR9_1_CTG3'\t138400797\tStandard\n",
      "#\t\t'HSCHR9_1_CTG6'\t138399532\tStandard\n",
      "#\t\t'HSCHR9_1_CTG7'\t138399111\tStandard\n",
      "#\t\t'HG613_PATCH'\t138394874\tStandard\n",
      "#\t\t'HG2030_PATCH'\t138394865\tStandard\n",
      "#\t\t'9'\t138394717\tStandard\n",
      "#\t\t'HSCHR11_2_CTG8'\t135117433\tStandard\n",
      "#\t\t'HG151_NOVEL_TEST'\t135097696\tStandard\n",
      "#\t\t'HSCHR11_1_CTG5'\t135097255\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_2'\t135096748\tStandard\n",
      "#\t\t'HG142_HG150_NOVEL_TEST'\t135094228\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1'\t135093444\tStandard\n",
      "#\t\t'HSCHR11_1_CTG1_1'\t135092355\tStandard\n",
      "#\t\t'HSCHR11_1_CTG6'\t135092332\tStandard\n",
      "#\t\t'HSCHR11_3_CTG1'\t135092180\tStandard\n",
      "#\t\t'HG2115_PATCH'\t135092089\tStandard\n",
      "#\t\t'HG2114_PATCH'\t135091163\tStandard\n",
      "#\t\t'HSCHR11_1_CTG3_1'\t135090845\tStandard\n",
      "#\t\t'HG2116_PATCH'\t135090698\tStandard\n",
      "#\t\t'HSCHR11_1_CTG3'\t135090275\tStandard\n",
      "#\t\t'HG2111_PATCH'\t135089248\tStandard\n",
      "#\t\t'HG28_PATCH'\t135088641\tStandard\n",
      "#\t\t'HG107_PATCH'\t135088590\tStandard\n",
      "#\t\t'HSCHR11_1_CTG8'\t135088433\tStandard\n",
      "#\t\t'HG1521_PATCH'\t135087279\tStandard\n",
      "#\t\t'11'\t135086622\tStandard\n",
      "#\t\t'HG2217_PATCH'\t135086621\tStandard\n",
      "#\t\t'HG1708_PATCH'\t135086620\tStandard\n",
      "#\t\t'HG2060_PATCH'\t135086593\tStandard\n",
      "#\t\t'HSCHR11_1_CTG7'\t135085484\tStandard\n",
      "#\t\t'HSCHR11_2_CTG1_1'\t135083841\tStandard\n",
      "#\t\t'HSCHR11_1_CTG2'\t135083598\tStandard\n",
      "#\t\t'HG1445_PATCH'\t135063159\tStandard\n",
      "#\t\t'HSCHR10_1_CTG1'\t133844722\tStandard\n",
      "#\t\t'HG545_PATCH'\t133833661\tStandard\n",
      "#\t\t'HSCHR10_1_CTG4'\t133815819\tStandard\n",
      "#\t\t'HSCHR10_1_CTG2'\t133813485\tStandard\n",
      "#\t\t'HSCHR10_1_CTG6'\t133801239\tStandard\n",
      "#\t\t'HG2191_PATCH'\t133799479\tStandard\n",
      "#\t\t'HG2244_HG2245_PATCH'\t133797425\tStandard\n",
      "#\t\t'10'\t133797422\tStandard\n",
      "#\t\t'HG2334_PATCH'\t133797389\tStandard\n",
      "#\t\t'HG2242_HG2243_PATCH'\t133797297\tStandard\n",
      "#\t\t'HG2241_PATCH'\t133792722\tStandard\n",
      "#\t\t'HSCHR10_1_CTG3'\t133783218\tStandard\n",
      "#\t\t'HG1277_PATCH'\t133754853\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2'\t133307257\tStandard\n",
      "#\t\t'HSCHR12_9_CTG2_1'\t133289677\tStandard\n",
      "#\t\t'HSCHR12_7_CTG2_1'\t133289478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2_1'\t133284468\tStandard\n",
      "#\t\t'HSCHR12_8_CTG2_1'\t133282605\tStandard\n",
      "#\t\t'HSCHR12_2_CTG2_1'\t133282358\tStandard\n",
      "#\t\t'HSCHR12_2_CTG1'\t133282237\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2_1'\t133281486\tStandard\n",
      "#\t\t'HSCHR12_1_CTG2'\t133281368\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2_1'\t133281351\tStandard\n",
      "#\t\t'HG1815_PATCH'\t133281281\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2_1'\t133279478\tStandard\n",
      "#\t\t'HSCHR12_1_CTG1'\t133279324\tStandard\n",
      "#\t\t'HSCHR12_4_CTG2'\t133277582\tStandard\n",
      "#\t\t'HSCHR12_6_CTG2_1'\t133275361\tStandard\n",
      "#\t\t'HG1362_PATCH'\t133275314\tStandard\n",
      "#\t\t'12'\t133275309\tStandard\n",
      "#\t\t'HG2047_PATCH'\t133275304\tStandard\n",
      "#\t\t'HSCHR12_5_CTG2'\t133275237\tStandard\n",
      "#\t\t'HG23_PATCH'\t133275184\tStandard\n",
      "#\t\t'HG2063_PATCH'\t133275140\tStandard\n",
      "#\t\t'HG2246_HG2248_HG2276_PATCH'\t133274640\tStandard\n",
      "#\t\t'HG1398_PATCH'\t133274540\tStandard\n",
      "#\t\t'HG2247_PATCH'\t133274000\tStandard\n",
      "#\t\t'HSCHR12_3_CTG2'\t133271566\tStandard\n",
      "#\t\t'HG2291_PATCH'\t114534236\tStandard\n",
      "#\t\t'HG1523_PATCH'\t114458500\tStandard\n",
      "#\t\t'HG2216_PATCH'\t114376428\tStandard\n",
      "#\t\t'HSCHR13_1_CTG7'\t114374491\tStandard\n",
      "#\t\t'HSCHR13_1_CTG3'\t114370226\tStandard\n",
      "#\t\t'HG1817_1_PATCH'\t114369395\tStandard\n",
      "#\t\t'HSCHR13_1_CTG8'\t114368996\tStandard\n",
      "#\t\t'HSCHR13_1_CTG4'\t114368332\tStandard\n",
      "#\t\t'HSCHR13_1_CTG2'\t114368315\tStandard\n",
      "#\t\t'HSCHR13_1_CTG6'\t114367169\tStandard\n",
      "#\t\t'HSCHR13_1_CTG5'\t114365350\tStandard\n",
      "#\t\t'13'\t114364328\tStandard\n",
      "#\t\t'HG2509_PATCH'\t114364327\tStandard\n",
      "#\t\t'HG2249_PATCH'\t114364211\tStandard\n",
      "#\t\t'HG2288_HG2289_PATCH'\t114360394\tStandard\n",
      "#\t\t'HSCHR13_1_CTG1'\t114348172\tStandard\n",
      "#\t\t'HG1524_PATCH'\t114315104\tStandard\n",
      "#\t\t'HSCHR14_3_CTG1'\t107067894\tStandard\n",
      "#\t\t'HSCHR14_2_CTG1'\t107050862\tStandard\n",
      "#\t\t'HSCHR14_8_CTG1'\t107050407\tStandard\n",
      "#\t\t'HSCHR14_9_CTG1'\t107050362\tStandard\n",
      "#\t\t'HSCHR14_1_CTG1'\t107047377\tStandard\n",
      "#\t\t'HSCHR14_7_CTG1'\t107045728\tStandard\n",
      "#\t\t'14'\t107043718\tStandard\n",
      "#\t\t'HG1_PATCH'\t107043717\tStandard\n",
      "#\t\t'HG2510_PATCH'\t107043717\tStandard\n",
      "#\t\t'HG2365_PATCH'\t102714182\tStandard\n",
      "#\t\t'HSCHR15_4_CTG8'\t102071387\tStandard\n",
      "#\t\t'HSCHR15_5_CTG8'\t102070278\tStandard\n",
      "#\t\t'HSCHR15_2_CTG8'\t102003326\tStandard\n",
      "#\t\t'HSCHR15_1_CTG8'\t102002809\tStandard\n",
      "#\t\t'HG2499_PATCH'\t102001626\tStandard\n",
      "#\t\t'HSCHR15_3_CTG8'\t101999716\tStandard\n",
      "#\t\t'HSCHR15_1_CTG1'\t101998275\tStandard\n",
      "#\t\t'HSCHR15_2_CTG3'\t101995966\tStandard\n",
      "#\t\t'HSCHR15_1_CTG3'\t101994657\tStandard\n",
      "#\t\t'HSCHR15_3_CTG3'\t101993763\tStandard\n",
      "#\t\t'HG2198_PATCH'\t101993033\tStandard\n",
      "#\t\t'15'\t101991189\tStandard\n",
      "#\t\t'HG2511_PATCH'\t101991188\tStandard\n",
      "#\t\t'HSCHR15_6_CTG8'\t101521929\tStandard\n",
      "#\t\t'HG926_PATCH'\t90680781\tStandard\n",
      "#\t\t'HSCHR16_1_CTG1'\t90426449\tStandard\n",
      "#\t\t'HSCHR16_3_CTG3_1'\t90346170\tStandard\n",
      "#\t\t'HSCHR16_5_CTG1'\t90344275\tStandard\n",
      "#\t\t'HSCHR16_2_CTG3_1'\t90343477\tStandard\n",
      "#\t\t'HSCHR16_4_CTG3_1'\t90343133\tStandard\n",
      "#\t\t'HSCHR16_CTG2'\t90341867\tStandard\n",
      "#\t\t'HG2471_PATCH'\t90341128\tStandard\n",
      "#\t\t'HSCHR16_4_CTG1'\t90340369\tStandard\n",
      "#\t\t'HSCHR16_3_CTG1'\t90340304\tStandard\n",
      "#\t\t'HG2263_PATCH'\t90338583\tStandard\n",
      "#\t\t'HSCHR16_5_CTG3_1'\t90338377\tStandard\n",
      "#\t\t'16'\t90338345\tStandard\n",
      "#\t\t'HSCHR16_1_CTG3_1'\t90335239\tStandard\n",
      "#\t\t'HSCHR17_1_CTG5'\t83552665\tStandard\n",
      "#\t\t'HSCHR17_10_CTG4'\t83359087\tStandard\n",
      "#\t\t'HSCHR17_1_CTG1'\t83290495\tStandard\n",
      "#\t\t'HSCHR17_12_CTG4'\t83271781\tStandard\n",
      "#\t\t'HSCHR17_3_CTG4'\t83269063\tStandard\n",
      "#\t\t'HSCHR17_3_CTG1'\t83267046\tStandard\n",
      "#\t\t'HSCHR17_11_CTG4'\t83264946\tStandard\n",
      "#\t\t'HSCHR17_1_CTG4'\t83264766\tStandard\n",
      "#\t\t'HSCHR17_9_CTG4'\t83263791\tStandard\n",
      "#\t\t'HSCHR17_2_CTG4'\t83263494\tStandard\n",
      "#\t\t'HG2087_PATCH'\t83260478\tStandard\n",
      "#\t\t'HSCHR17_2_CTG1'\t83259496\tStandard\n",
      "#\t\t'HSCHR17_8_CTG4'\t83258474\tStandard\n",
      "#\t\t'HSCHR17_3_CTG2'\t83258211\tStandard\n",
      "#\t\t'HSCHR17_4_CTG4'\t83257896\tStandard\n",
      "#\t\t'17'\t83257441\tStandard\n",
      "#\t\t'HG2046_PATCH'\t83257439\tStandard\n",
      "#\t\t'HSCHR17_1_CTG2'\t83256362\tStandard\n",
      "#\t\t'HSCHR17_1_CTG9'\t83255504\tStandard\n",
      "#\t\t'HSCHR17_5_CTG4'\t83247119\tStandard\n",
      "#\t\t'HSCHR17_6_CTG4'\t83246071\tStandard\n",
      "#\t\t'HG2285_HG106_HG2252_PATCH'\t83219555\tStandard\n",
      "#\t\t'HG1320_PATCH'\t83207673\tStandard\n",
      "#\t\t'HSCHR17_2_CTG2'\t83160220\tStandard\n",
      "#\t\t'HSCHR17_2_CTG5'\t83086355\tStandard\n",
      "#\t\t'HSCHR17_7_CTG4'\t83058228\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2'\t80442734\tStandard\n",
      "#\t\t'HSCHR18_1_CTG1'\t80395815\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2'\t80386177\tStandard\n",
      "#\t\t'HSCHR18_1_CTG2_1'\t80382671\tStandard\n",
      "#\t\t'HSCHR18_1_CTG1_1'\t80380517\tStandard\n",
      "#\t\t'HSCHR18_2_CTG1_1'\t80379367\tStandard\n",
      "#\t\t'HG2442_PATCH'\t80379182\tStandard\n",
      "#\t\t'HSCHR18_3_CTG2_1'\t80379039\tStandard\n",
      "#\t\t'HSCHR18_5_CTG1_1'\t80378965\tStandard\n",
      "#\t\t'HSCHR18_2_CTG2_1'\t80378720\tStandard\n",
      "#\t\t'HSCHR18_4_CTG1_1'\t80378406\tStandard\n",
      "#\t\t'HG2213_PATCH'\t80377536\tStandard\n",
      "#\t\t'HSCHR18_ALT2_CTG2_1'\t80376883\tStandard\n",
      "#\t\t'HSCHR18_ALT21_CTG2_1'\t80376343\tStandard\n",
      "#\t\t'18'\t80373285\tStandard\n",
      "#\t\t'HG2412_PATCH'\t80373284\tStandard\n",
      "#\t\t'HSCHR20_1_CTG1'\t64450183\tStandard\n",
      "#\t\t'HSCHR20_1_CTG3'\t64447812\tStandard\n",
      "#\t\t'HSCHR20_1_CTG4'\t64446453\tStandard\n",
      "#\t\t'20'\t64444167\tStandard\n",
      "#\t\t'HSCHR20_1_CTG2'\t64441059\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TB04_CTG3_1'\t58750420\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_B_HAP_CTG3_1'\t58689284\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_B_HAP_CTG3_1'\t58689016\tStandard\n",
      "#\t\t'HSCHR19KIR_0019-4656-B_CTG3_1'\t58675214\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_BA2_HAP_CTG3_1'\t58662330\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_BA2_HAP_CTG3_1'\t58662237\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_BA2_HAP_CTG3_1'\t58661364\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_J_CTG3_1'\t58650772\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_BAX_HAP_CTG3_1'\t58647709\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TB01_CTG3_1'\t58645065\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_AB_HAP_CTG3_1'\t58644315\tStandard\n",
      "#\t\t'HG2021_PATCH'\t58632883\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_BDEL_HAP_CTG3_1'\t58631446\tStandard\n",
      "#\t\t'HSCHR19KIR_RP5_B_HAP_CTG3_1'\t58631402\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_BDEL_HAP_CTG3_1'\t58631206\tStandard\n",
      "#\t\t'HSCHR19_2_CTG2'\t58627095\tStandard\n",
      "#\t\t'HSCHR19_3_CTG3_1'\t58625958\tStandard\n",
      "#\t\t'HSCHR19_3_CTG2'\t58625823\tStandard\n",
      "#\t\t'HSCHR19_4_CTG3_1'\t58625731\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_S_CTG3_1'\t58625321\tStandard\n",
      "#\t\t'HSCHR19_1_CTG3_1'\t58624789\tStandard\n",
      "#\t\t'HSCHR19_1_CTG2'\t58624032\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_I_CTG3_1'\t58623235\tStandard\n",
      "#\t\t'HSCHR19_4_CTG2'\t58619803\tStandard\n",
      "#\t\t'HSCHR19_5_CTG2'\t58619095\tStandard\n",
      "#\t\t'HSCHR19_2_CTG3_1'\t58618601\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_C_P_CTG3_1'\t58618008\tStandard\n",
      "#\t\t'HG109_PATCH'\t58617934\tStandard\n",
      "#\t\t'HSCHR19KIR_FH08_A_HAP_CTG3_1'\t58617658\tStandard\n",
      "#\t\t'HG26_PATCH'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_G248_A_HAP_CTG3_1'\t58617652\tStandard\n",
      "#\t\t'HSCHR19KIR_7191059-1_CTG3_1'\t58617649\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TA01_2_CTG3_1'\t58617647\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_AB_HAP_T_P_CTG3_1'\t58617636\tStandard\n",
      "#\t\t'HSCHR19KIR_LUCE_A_HAP_CTG3_1'\t58617634\tStandard\n",
      "#\t\t'HSCHR19KIR_CA01-TA01_1_CTG3_1'\t58617632\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2394_CTG3_1'\t58617630\tStandard\n",
      "#\t\t'HSCHR19KIR_ABC08_A1_HAP_CTG3_1'\t58617625\tStandard\n",
      "#\t\t'19'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_RSH_A_HAP_CTG3_1'\t58617616\tStandard\n",
      "#\t\t'HSCHR19KIR_502960008-1_CTG3_1'\t58617609\tStandard\n",
      "#\t\t'HSCHR19KIR_7191059-2_CTG3_1'\t58617604\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_T7526_A_HAP_CTG3_1'\t58617601\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_A_HAP_CTG3_1'\t58617587\tStandard\n",
      "#\t\t'HSCHR19KIR_FH15_A_HAP_CTG3_1'\t58617473\tStandard\n",
      "#\t\t'HSCHR19KIR_0019-4656-A_CTG3_1'\t58617441\tStandard\n",
      "#\t\t'HSCHR19KIR_FH05_A_HAP_CTG3_1'\t58616795\tStandard\n",
      "#\t\t'HSCHR19KIR_FH13_A_HAP_CTG3_1'\t58616725\tStandard\n",
      "#\t\t'HSCHR19KIR_502960008-2_CTG3_1'\t58616590\tStandard\n",
      "#\t\t'HSCHR19KIR_FH06_BA1_HAP_CTG3_1'\t58603994\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2396_CTG3_1'\t58603982\tStandard\n",
      "#\t\t'HSCHR19KIR_G085_BA1_HAP_CTG3_1'\t58603800\tStandard\n",
      "#\t\t'HSCHR19KIR_GRC212_BA1_HAP_CTG3_1'\t58602595\tStandard\n",
      "#\t\t'HSCHR19KIR_HG2393_CTG3_1'\t58583172\tStandard\n",
      "#\t\t'HSCHR19LRC_LRC_T_CTG3_1'\t58561614\tStandard\n",
      "#\t\t'HSCHR19LRC_COX1_CTG3_1'\t58546647\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF1_CTG3_1'\t58546031\tStandard\n",
      "#\t\t'HSCHR19KIR_0010-5217-AB_CTG3_1'\t58538377\tStandard\n",
      "#\t\t'HSCHR19KIR_CA04_CTG3_1'\t58537114\tStandard\n",
      "#\t\t'HSCHR19LRC_PGF2_CTG3_1'\t58355410\tStandard\n",
      "#\t\t'HSCHR19LRC_COX2_CTG3_1'\t58288451\tStandard\n",
      "#\t\t'HG2062_PATCH'\t57262172\tStandard\n",
      "#\t\t'Y'\t57227415\tStandard\n",
      "#\t\t'HG1535_PATCH'\t57187419\tStandard\n",
      "#\t\t'HG1531_PATCH'\t57177176\tStandard\n",
      "#\t\t'HG1485_PATCH'\t50915261\tStandard\n",
      "#\t\t'HSCHR22_1_CTG7'\t50872732\tStandard\n",
      "#\t\t'HSCHR22_7_CTG1'\t50845852\tStandard\n",
      "#\t\t'HSCHR22_1_CTG6'\t50842710\tStandard\n",
      "#\t\t'HSCHR22_5_CTG1'\t50832153\tStandard\n",
      "#\t\t'HSCHR22_8_CTG1'\t50832130\tStandard\n",
      "#\t\t'HSCHR22_2_CTG1'\t50830603\tStandard\n",
      "#\t\t'HSCHR22_1_CTG3'\t50827573\tStandard\n",
      "#\t\t'HSCHR22_1_CTG4'\t50824699\tStandard\n",
      "#\t\t'HSCHR22_1_CTG5'\t50821308\tStandard\n",
      "#\t\t'HG1311_PATCH'\t50818486\tStandard\n",
      "#\t\t'HG494_PATCH'\t50818473\tStandard\n",
      "#\t\t'22'\t50818468\tStandard\n",
      "#\t\t'HSCHR22_3_CTG1'\t50818468\tStandard\n",
      "#\t\t'HG2512_PATCH'\t50818467\tStandard\n",
      "#\t\t'HSCHR22_6_CTG1'\t50815919\tStandard\n",
      "#\t\t'HSCHR22_1_CTG1'\t50806372\tStandard\n",
      "#\t\t'HSCHR22_4_CTG1'\t50806321\tStandard\n",
      "#\t\t'HSCHR22_1_CTG2'\t50788525\tStandard\n",
      "#\t\t'HG2513_PATCH'\t47068079\tStandard\n",
      "#\t\t'HSCHR21_1_CTG1_1'\t46722569\tStandard\n",
      "#\t\t'HSCHR21_4_CTG1_1'\t46719541\tStandard\n",
      "#\t\t'HSCHR21_2_CTG1_1'\t46716685\tStandard\n",
      "#\t\t'HSCHR21_3_CTG1_1'\t46715777\tStandard\n",
      "#\t\t'HSCHR21_6_CTG1_1'\t46713516\tStandard\n",
      "#\t\t'HSCHR21_8_CTG1_1'\t46713185\tStandard\n",
      "#\t\t'HSCHR21_5_CTG2'\t46712294\tStandard\n",
      "#\t\t'21'\t46709983\tStandard\n",
      "#\t\t'KI270728.1'\t1872759\tStandard\n",
      "#\t\t'KI270727.1'\t448248\tStandard\n",
      "#\t\t'KI270442.1'\t392061\tStandard\n",
      "#\t\t'KI270729.1'\t280839\tStandard\n",
      "#\t\t'GL000225.1'\t211173\tStandard\n",
      "#\t\t'KI270743.1'\t210658\tStandard\n",
      "#\t\t'GL000008.2'\t209709\tStandard\n",
      "#\t\t'GL000009.2'\t201709\tStandard\n",
      "#\t\t'KI270747.1'\t198735\tStandard\n",
      "#\t\t'KI270722.1'\t194050\tStandard\n",
      "#\t\t'GL000194.1'\t191469\tStandard\n",
      "#\t\t'KI270742.1'\t186739\tStandard\n",
      "#\t\t'GL000205.2'\t185591\tStandard\n",
      "#\t\t'GL000195.1'\t182896\tStandard\n",
      "#\t\t'KI270736.1'\t181920\tStandard\n",
      "#\t\t'KI270733.1'\t179772\tStandard\n",
      "#\t\t'GL000224.1'\t179693\tStandard\n",
      "#\t\t'GL000219.1'\t179198\tStandard\n",
      "#\t\t'KI270719.1'\t176845\tStandard\n",
      "#\t\t'GL000216.2'\t176608\tStandard\n",
      "#\t\t'KI270712.1'\t176043\tStandard\n",
      "#\t\t'KI270706.1'\t175055\tStandard\n",
      "#\t\t'KI270725.1'\t172810\tStandard\n",
      "#\t\t'KI270744.1'\t168472\tStandard\n",
      "#\t\t'KI270734.1'\t165050\tStandard\n",
      "#\t\t'GL000213.1'\t164239\tStandard\n",
      "#\t\t'GL000220.1'\t161802\tStandard\n",
      "#\t\t'KI270715.1'\t161471\tStandard\n",
      "#\t\t'GL000218.1'\t161147\tStandard\n",
      "#\t\t'KI270749.1'\t158759\tStandard\n",
      "#\t\t'KI270741.1'\t157432\tStandard\n",
      "#\t\t'GL000221.1'\t155397\tStandard\n",
      "#\t\t'KI270716.1'\t153799\tStandard\n",
      "#\t\t'KI270731.1'\t150754\tStandard\n",
      "#\t\t'KI270751.1'\t150742\tStandard\n",
      "#\t\t'KI270750.1'\t148850\tStandard\n",
      "#\t\t'KI270519.1'\t138126\tStandard\n",
      "#\t\t'GL000214.1'\t137718\tStandard\n",
      "#\t\t'KI270708.1'\t127682\tStandard\n",
      "#\t\t'KI270730.1'\t112551\tStandard\n",
      "#\t\t'KI270438.1'\t112505\tStandard\n",
      "#\t\t'KI270737.1'\t103838\tStandard\n",
      "#\t\t'KI270721.1'\t100316\tStandard\n",
      "#\t\t'KI270738.1'\t99375\tStandard\n",
      "#\t\t'KI270748.1'\t93321\tStandard\n",
      "#\t\t'KI270435.1'\t92983\tStandard\n",
      "#\t\t'GL000208.1'\t92689\tStandard\n",
      "#\t\t'KI270538.1'\t91309\tStandard\n",
      "#\t\t'KI270756.1'\t79590\tStandard\n",
      "#\t\t'KI270739.1'\t73985\tStandard\n",
      "#\t\t'KI270757.1'\t71251\tStandard\n",
      "#\t\t'KI270709.1'\t66860\tStandard\n",
      "#\t\t'KI270746.1'\t66486\tStandard\n",
      "#\t\t'KI270753.1'\t62944\tStandard\n",
      "#\t\t'KI270589.1'\t44474\tStandard\n",
      "#\t\t'KI270726.1'\t43739\tStandard\n",
      "#\t\t'KI270735.1'\t42811\tStandard\n",
      "#\t\t'KI270711.1'\t42210\tStandard\n",
      "#\t\t'KI270745.1'\t41891\tStandard\n",
      "#\t\t'KI270714.1'\t41717\tStandard\n",
      "#\t\t'KI270732.1'\t41543\tStandard\n",
      "#\t\t'KI270713.1'\t40745\tStandard\n",
      "#\t\t'KI270754.1'\t40191\tStandard\n",
      "#\t\t'KI270710.1'\t40176\tStandard\n",
      "#\t\t'KI270717.1'\t40062\tStandard\n",
      "#\t\t'KI270724.1'\t39555\tStandard\n",
      "#\t\t'KI270720.1'\t39050\tStandard\n",
      "#\t\t'KI270723.1'\t38115\tStandard\n",
      "#\t\t'KI270718.1'\t38054\tStandard\n",
      "#\t\t'KI270317.1'\t37690\tStandard\n",
      "#\t\t'KI270740.1'\t37240\tStandard\n",
      "#\t\t'KI270755.1'\t36723\tStandard\n",
      "#\t\t'KI270707.1'\t32032\tStandard\n",
      "#\t\t'KI270579.1'\t31033\tStandard\n",
      "#\t\t'KI270752.1'\t27745\tStandard\n",
      "#\t\t'KI270512.1'\t22689\tStandard\n",
      "#\t\t'KI270322.1'\t21476\tStandard\n",
      "#\t\t'MT'\t16569\tStandard\n",
      "#\t\t'GL000226.1'\t15008\tStandard\n",
      "#\t\t'KI270311.1'\t12399\tStandard\n",
      "#\t\t'KI270366.1'\t8320\tStandard\n",
      "#\t\t'KI270511.1'\t8127\tStandard\n",
      "#\t\t'KI270448.1'\t7992\tStandard\n",
      "#\t\t'KI270521.1'\t7642\tStandard\n",
      "#\t\t'KI270581.1'\t7046\tStandard\n",
      "#\t\t'KI270582.1'\t6504\tStandard\n",
      "#\t\t'KI270515.1'\t6361\tStandard\n",
      "#\t\t'KI270588.1'\t6158\tStandard\n",
      "#\t\t'KI270591.1'\t5796\tStandard\n",
      "#\t\t'KI270522.1'\t5674\tStandard\n",
      "#\t\t'KI270507.1'\t5353\tStandard\n",
      "#\t\t'KI270590.1'\t4685\tStandard\n",
      "#\t\t'KI270584.1'\t4513\tStandard\n",
      "#\t\t'KI270320.1'\t4416\tStandard\n",
      "#\t\t'KI270382.1'\t4215\tStandard\n",
      "#\t\t'KI270468.1'\t4055\tStandard\n",
      "#\t\t'KI270467.1'\t3920\tStandard\n",
      "#\t\t'KI270362.1'\t3530\tStandard\n",
      "#\t\t'KI270517.1'\t3253\tStandard\n",
      "#\t\t'KI270593.1'\t3041\tStandard\n",
      "#\t\t'KI270528.1'\t2983\tStandard\n",
      "#\t\t'KI270587.1'\t2969\tStandard\n",
      "#\t\t'KI270364.1'\t2855\tStandard\n",
      "#\t\t'KI270371.1'\t2805\tStandard\n",
      "#\t\t'KI270333.1'\t2699\tStandard\n",
      "#\t\t'KI270374.1'\t2656\tStandard\n",
      "#\t\t'KI270411.1'\t2646\tStandard\n",
      "#\t\t'KI270414.1'\t2489\tStandard\n",
      "#\t\t'KI270510.1'\t2415\tStandard\n",
      "#\t\t'KI270390.1'\t2387\tStandard\n",
      "#\t\t'KI270375.1'\t2378\tStandard\n",
      "#\t\t'KI270420.1'\t2321\tStandard\n",
      "#\t\t'KI270509.1'\t2318\tStandard\n",
      "#\t\t'KI270315.1'\t2276\tStandard\n",
      "#\t\t'KI270302.1'\t2274\tStandard\n",
      "#\t\t'KI270518.1'\t2186\tStandard\n",
      "#\t\t'KI270530.1'\t2168\tStandard\n",
      "#\t\t'KI270304.1'\t2165\tStandard\n",
      "#\t\t'KI270418.1'\t2145\tStandard\n",
      "#\t\t'KI270424.1'\t2140\tStandard\n",
      "#\t\t'KI270417.1'\t2043\tStandard\n",
      "#\t\t'KI270508.1'\t1951\tStandard\n",
      "#\t\t'KI270303.1'\t1942\tStandard\n",
      "#\t\t'KI270381.1'\t1930\tStandard\n",
      "#\t\t'KI270529.1'\t1899\tStandard\n",
      "#\t\t'KI270425.1'\t1884\tStandard\n",
      "#\t\t'KI270396.1'\t1880\tStandard\n",
      "#\t\t'KI270363.1'\t1803\tStandard\n",
      "#\t\t'KI270386.1'\t1788\tStandard\n",
      "#\t\t'KI270465.1'\t1774\tStandard\n",
      "#\t\t'KI270383.1'\t1750\tStandard\n",
      "#\t\t'KI270384.1'\t1658\tStandard\n",
      "#\t\t'KI270330.1'\t1652\tStandard\n",
      "#\t\t'KI270372.1'\t1650\tStandard\n",
      "#\t\t'KI270548.1'\t1599\tStandard\n",
      "#\t\t'KI270580.1'\t1553\tStandard\n",
      "#\t\t'KI270387.1'\t1537\tStandard\n",
      "#\t\t'KI270391.1'\t1484\tStandard\n",
      "#\t\t'KI270305.1'\t1472\tStandard\n",
      "#\t\t'KI270373.1'\t1451\tStandard\n",
      "#\t\t'KI270422.1'\t1445\tStandard\n",
      "#\t\t'KI270316.1'\t1444\tStandard\n",
      "#\t\t'KI270338.1'\t1428\tStandard\n",
      "#\t\t'KI270340.1'\t1428\tStandard\n",
      "#\t\t'KI270583.1'\t1400\tStandard\n",
      "#\t\t'KI270334.1'\t1368\tStandard\n",
      "#\t\t'KI270429.1'\t1361\tStandard\n",
      "#\t\t'KI270393.1'\t1308\tStandard\n",
      "#\t\t'KI270516.1'\t1300\tStandard\n",
      "#\t\t'KI270389.1'\t1298\tStandard\n",
      "#\t\t'KI270466.1'\t1233\tStandard\n",
      "#\t\t'KI270388.1'\t1216\tStandard\n",
      "#\t\t'KI270544.1'\t1202\tStandard\n",
      "#\t\t'KI270310.1'\t1201\tStandard\n",
      "#\t\t'KI270412.1'\t1179\tStandard\n",
      "#\t\t'KI270395.1'\t1143\tStandard\n",
      "#\t\t'KI270376.1'\t1136\tStandard\n",
      "#\t\t'KI270337.1'\t1121\tStandard\n",
      "#\t\t'KI270335.1'\t1048\tStandard\n",
      "#\t\t'KI270378.1'\t1048\tStandard\n",
      "#\t\t'KI270379.1'\t1045\tStandard\n",
      "#\t\t'KI270329.1'\t1040\tStandard\n",
      "#\t\t'KI270419.1'\t1029\tStandard\n",
      "#\t\t'KI270336.1'\t1026\tStandard\n",
      "#\t\t'KI270312.1'\t998\tStandard\n",
      "#\t\t'KI270539.1'\t993\tStandard\n",
      "#\t\t'KI270385.1'\t990\tStandard\n",
      "#\t\t'KI270423.1'\t981\tStandard\n",
      "#\t\t'KI270392.1'\t971\tStandard\n",
      "#\t\t'KI270394.1'\t970\tStandard\n",
      "#-----------------------------------------------\n",
      "\n",
      "00:00:23 Predicting variants\n",
      "00:00:23 Loading sequences for chromosome '12' from file '/home/mgpersonal/miniconda3/envs/ngs/share/snpeff-5.3.0a-1/./data/GRCh38.99/sequence.12.bin'\n",
      "00:00:24 Building sequence tree for chromosome '12'\n",
      "00:00:24 Done. Loaded 1488 sequences.\n",
      "\n",
      "WARNINGS: Some warning were detected\n",
      "Warning type\tNumber of warnings\n",
      "INFO_REALIGN_3_PRIME\t4\n",
      "WARNING_TRANSCRIPT_NO_STOP_CODON\t3\n",
      "\n",
      "\n",
      "00:00:25 Creating summary file: snpEff_summary.html\n",
      "00:00:25 Creating genes file: snpEff_genes.txt\n",
      "00:00:25 done.\n",
      "00:00:25 Done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "set -euo pipefail\n",
    "cd ~/ctdna_kras_project/results/vcf\n",
    "\n",
    "# baseline\n",
    "snpEff -Xmx4g -v GRCh38.99 \\\n",
    "  C_fw075_baseline.kras.varscan.vcf \\\n",
    "  > C_fw075_baseline.kras.ann.vcf\n",
    "\n",
    "# follow-up\n",
    "snpEff -Xmx4g -v GRCh38.99 \\\n",
    "  C_fw075_followup.kras.varscan.vcf \\\n",
    "  > C_fw075_followup.kras.ann.vcf\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "68d8c55a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mgpersonal/ctdna_kras_project/results/vcf\n",
      "total 192M\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 1.2M Nov 15 15:01 C_fw075_baseline.kras.ann.vcf\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal    0 Nov 14 21:42 C_fw075_baseline.kras.clinvar.tsv\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 5.7M Nov 15 15:00 C_fw075_baseline.kras.mpileup\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 460K Nov 14 22:29 C_fw075_baseline.kras.summary.tsv\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 1.2M Nov 15 15:00 C_fw075_baseline.kras.varscan.vcf\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 1.2M Nov 15 15:01 C_fw075_followup.kras.ann.vcf\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 7.2M Nov 15 15:00 C_fw075_followup.kras.mpileup\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 420K Nov 14 22:29 C_fw075_followup.kras.summary.tsv\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 1.2M Nov 15 15:00 C_fw075_followup.kras.varscan.vcf\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 173M Nov  9 14:34 clinvar.vcf.gz\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 592K Nov  9 14:34 clinvar.vcf.gz.tbi\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal 1.3K Nov 15 15:01 snpEff_genes.txt\n",
      "-rw-r--r-- 1 mgpersonal mgpersonal  23K Nov 15 15:01 snpEff_summary.html\n"
     ]
    }
   ],
   "source": [
    "\n",
    "%%bash\n",
    "set -euo pipefail\n",
    "cd ~/ctdna_kras_project/results/vcf\n",
    "pwd\n",
    "ls -lh\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "16e6c4c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Baseline KRAS coding/splice variants with impact:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF_COUNT</th>\n",
       "      <th>ALT_COUNT</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>VAF_percent</th>\n",
       "      <th>cDNA_change</th>\n",
       "      <th>protein_change</th>\n",
       "      <th>effect</th>\n",
       "      <th>impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>25245350</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1201</td>\n",
       "      <td>110</td>\n",
       "      <td>1315</td>\n",
       "      <td>8.37</td>\n",
       "      <td>c.35G&gt;A</td>\n",
       "      <td>p.Gly12Asp</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  REF_COUNT  ALT_COUNT  DEPTH  VAF_percent  \\\n",
       "0    12  25245350   C   T       1201        110   1315         8.37   \n",
       "\n",
       "  cDNA_change protein_change            effect    impact  \n",
       "0     c.35G>A     p.Gly12Asp  missense_variant  MODERATE  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Follow-up KRAS coding/splice variants with impact:\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>REF_COUNT</th>\n",
       "      <th>ALT_COUNT</th>\n",
       "      <th>DEPTH</th>\n",
       "      <th>VAF_percent</th>\n",
       "      <th>cDNA_change</th>\n",
       "      <th>protein_change</th>\n",
       "      <th>effect</th>\n",
       "      <th>impact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12</td>\n",
       "      <td>25250747</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>44</td>\n",
       "      <td>2</td>\n",
       "      <td>46</td>\n",
       "      <td>4.35</td>\n",
       "      <td>c.-12+4C&gt;T</td>\n",
       "      <td></td>\n",
       "      <td>splice_region_variant&amp;intron_variant</td>\n",
       "      <td>LOW</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT  REF_COUNT  ALT_COUNT  DEPTH  VAF_percent  \\\n",
       "0    12  25250747   G   A         44          2     46         4.35   \n",
       "\n",
       "  cDNA_change protein_change                                effect impact  \n",
       "0  c.-12+4C>T                 splice_region_variant&intron_variant    LOW  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Make parsed summary tables from the annotated .vcfs\n",
    "\n",
    "baseline_vcf = os.path.join(vcf_dir, \"C_fw075_baseline.kras.ann.vcf\")\n",
    "followup_vcf  = os.path.join(vcf_dir, \"C_fw075_followup.kras.ann.vcf\")\n",
    "\n",
    "\n",
    "# Parse the SnpEff-annotated VCF and keep only KRAS variants that are missense/nonsense/splice-related and have a non-empty IMPACT field.\n",
    "def extract_kras_coding(vcf_path):\n",
    " \n",
    "    rows = []\n",
    "    with open(vcf_path) as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"#\"):\n",
    "                continue\n",
    "\n",
    "            parts = line.rstrip(\"\\n\").split(\"\\t\")\n",
    "            if len(parts) < 10:\n",
    "                continue\n",
    "\n",
    "            chrom, pos, vid, ref, alt, qual, flt, info, fmt, sample = parts\n",
    "\n",
    "            # Find ANN=... in INFO\n",
    "            ann = None\n",
    "            for kv in info.split(\";\"):\n",
    "                if kv.startswith(\"ANN=\"):\n",
    "                    ann = kv.split(\"=\", 1)[1]\n",
    "                    break\n",
    "            if ann is None:\n",
    "                continue\n",
    "\n",
    "            # Loop over each ANN entry. We just want effect, impact, gene, cdna, and protein to represent our findins.\n",
    "            for entry in ann.split(\",\"):\n",
    "                fields = entry.split(\"|\")\n",
    "                if len(fields) < 11:\n",
    "                    continue\n",
    "\n",
    "                effect  = fields[1]   \n",
    "                impact  = fields[2]   \n",
    "                gene    = fields[3]   \n",
    "                cdna    = fields[9]   \n",
    "                protein = fields[10]  \n",
    "\n",
    "                # Only KRAS\n",
    "                if gene != \"KRAS\":\n",
    "                    continue\n",
    "\n",
    "                # Require impact annotation\n",
    "                if impact in (\"\", \".\"):\n",
    "                    continue\n",
    "\n",
    "                # Only missense / nonsense / splice-related\n",
    "                effect_lower = effect.lower()\n",
    "                if not (\n",
    "                    \"missense\" in effect_lower\n",
    "                    or \"nonsense\" in effect_lower\n",
    "                    or \"splice\"   in effect_lower\n",
    "                ):\n",
    "                    continue\n",
    "\n",
    "                # Parse FORMAT from VarScan info\n",
    "                fmt_keys = fmt.split(\":\")\n",
    "                fmt_vals = sample.split(\":\")\n",
    "                fmt_dict = dict(zip(fmt_keys, fmt_vals))\n",
    "\n",
    "                rd = fmt_dict.get(\"RD\")\n",
    "                ad = fmt_dict.get(\"AD\")\n",
    "                dp = fmt_dict.get(\"DP\")\n",
    "                freq = fmt_dict.get(\"FREQ\")\n",
    "\n",
    "                try:\n",
    "                    rd_int = int(rd)\n",
    "                except:\n",
    "                    rd_int = np.nan\n",
    "\n",
    "                try:\n",
    "                    ad_int = int(ad)\n",
    "                except:\n",
    "                    ad_int = np.nan\n",
    "\n",
    "                try:\n",
    "                    dp_int = int(dp)\n",
    "                except:\n",
    "                    dp_int = rd_int + ad_int if not np.isnan(rd_int) else np.nan\n",
    "\n",
    "                vaf = np.nan\n",
    "                if freq and freq.endswith(\"%\"):\n",
    "                    try:\n",
    "                        vaf = float(freq.rstrip(\"%\"))\n",
    "                    except:\n",
    "                        vaf = np.nan\n",
    "\n",
    "                rows.append({\n",
    "                    \"CHROM\": chrom,\n",
    "                    \"POS\": int(pos),\n",
    "                    \"REF\": ref,\n",
    "                    \"ALT\": alt,\n",
    "                    \"REF_COUNT\": rd_int,\n",
    "                    \"ALT_COUNT\": ad_int,\n",
    "                    \"DEPTH\": dp_int,\n",
    "                    \"VAF_percent\": vaf,\n",
    "                    \"cDNA_change\": cdna,\n",
    "                    \"protein_change\": protein,\n",
    "                    \"effect\": effect,\n",
    "                    \"impact\": impact,\n",
    "                })\n",
    "\n",
    "    df = pd.DataFrame(rows)\n",
    "\n",
    "    # drop transcripts for summary\n",
    "    if not df.empty:\n",
    "        df = (\n",
    "            df.drop_duplicates(\n",
    "                subset=[\"CHROM\", \"POS\", \"REF\", \"ALT\", \"protein_change\", \"effect\", \"impact\"]\n",
    "            )\n",
    "            .reset_index(drop=True)\n",
    "        )\n",
    "\n",
    "    return df\n",
    "\n",
    "# Use the simpler extractor\n",
    "baseline_kras = extract_kras_coding(baseline_vcf)\n",
    "followup_kras = extract_kras_coding(followup_vcf)\n",
    "\n",
    "print(\"Baseline KRAS coding/splice variants with impact:\")\n",
    "display(baseline_kras)\n",
    "\n",
    "print(\"\\nFollow-up KRAS coding/splice variants with impact:\")\n",
    "display(followup_kras)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b8bb035",
   "metadata": {},
   "source": [
    "6. Ideas/Vision Board for visualizations for marketing or presentations\n",
    "\n",
    "With a variant calling analysis a lot of times its hard to avoid a polished summary table that shows relevant information like the Basline/follow-up tables above which ended our analysis. It is likely important for shareholders within the organization to see information like the depth and VAF, since these scientist understand all this information. For marketing to a physician or medical group it may be enough to just show the VAF. That being said, there are several figures that may communicate the table information well or support the table.\n",
    "\n",
    "1) In my analysis I only had a single post treatment or follow-up sample, but some patients had up to 4 that I saw. A nice longitudinal plot with the x axis the dates of sample procurement and y axis being the VAF may look nice. It could communicate response of the metastatic CRC to the cetuximab over those sample time points, or if it pops up again, perhaps resistance. Either way, showing an important hotspot mutation like KRAS G12D pre or post cetuximab would likely lead to getting the patient on a type 1 KRAS inhibitor. \n",
    "\n",
    "2) Imaging the KRAS alterations with IGV could be meaningful to internal shareholders and scientists since they understand more of the techanical and conceptual aspect of the NGS. I did not image this bam with IGV but from the depth of the KRAS G12D snp would would look striking. \n",
    "\n",
    "3) It may be nice for internal shareholders and marketing to show a nice EGFR and downstream MAPK pathway proteins. If we need to speak specifically about KRAS we could show how normal KRAS signals and how altered KRAS signals. It certainly would be nice to see where all the panel proteins are positioned in the signaling. \n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (ngs WSL)",
   "language": "python",
   "name": "ngs"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
